<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Fnu Avinesh Krishnan">
<meta name="dcterms.date" content="2023-05-22">
<meta name="description" content="Final Project writeup">

<title>Final Project: Fnu Avinesh Krishnan</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../images/DACSS_Round_Network.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="styles.css">
<meta name="twitter:title" content="Final Project: Fnu Avinesh Krishnan">
<meta name="twitter:description" content="Final Project writeup">
<meta name="twitter:image" content="https://dacss.github.io/601_Spring_2023/posts/FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-10-1.png">
<meta name="twitter:creator" content="@UMassDACSS">
<meta name="twitter:site" content="@UMassDACSS">
<meta name="twitter:image-height" content="960">
<meta name="twitter:image-width" content="1344">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../images/UMass White Wordmark Horiz.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/DACSS" rel="" target="">
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html" rel="" target="">
 <span class="menu-text">Contributors</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://umass.edu/sbs/dacss" rel="" target="">
 <span class="menu-text">DACSS</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/DACSS" rel="" target=""><i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#part-1.-introduction" id="toc-part-1.-introduction" class="nav-link active" data-scroll-target="#part-1.-introduction">Part 1. Introduction</a></li>
  <li><a href="#part-2.-background-of-the-topicliterature-review" id="toc-part-2.-background-of-the-topicliterature-review" class="nav-link" data-scroll-target="#part-2.-background-of-the-topicliterature-review">Part 2. Background of the Topic/Literature Review</a></li>
  <li><a href="#part-3.-dataset-introduction" id="toc-part-3.-dataset-introduction" class="nav-link" data-scroll-target="#part-3.-dataset-introduction">Part 3. Dataset Introduction</a></li>
  <li><a href="#part-4.-dataset-description" id="toc-part-4.-dataset-description" class="nav-link" data-scroll-target="#part-4.-dataset-description">Part 4. Dataset Description</a></li>
  <li><a href="#part-5.-data-pre-processing-and-analysis-plan" id="toc-part-5.-data-pre-processing-and-analysis-plan" class="nav-link" data-scroll-target="#part-5.-data-pre-processing-and-analysis-plan">Part 5. Data Pre-processing and Analysis Plan</a></li>
  <li><a href="#part-6.-data-pre-processing-and-visualization" id="toc-part-6.-data-pre-processing-and-visualization" class="nav-link" data-scroll-target="#part-6.-data-pre-processing-and-visualization">Part 6. Data Pre-processing and Visualization</a></li>
  <li><a href="#part-7.-conclusion-and-discussion" id="toc-part-7.-conclusion-and-discussion" class="nav-link" data-scroll-target="#part-7.-conclusion-and-discussion">Part 7. Conclusion and Discussion :</a></li>
  <li><a href="#part-8.-bibliography" id="toc-part-8.-bibliography" class="nav-link" data-scroll-target="#part-8.-bibliography">Part 8. Bibliography :</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Final Project: Fnu Avinesh Krishnan</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
  <div class="quarto-categories">
    <div class="quarto-category">final_Project</div>
  </div>
  </div>

<div>
  <div class="description">
    Final Project writeup
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Fnu Avinesh Krishnan </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 22, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">message=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="part-1.-introduction" class="level2">
<h2 class="anchored" data-anchor-id="part-1.-introduction">Part 1. Introduction</h2>
<p>Project Topic: The project aims to analyze factors of a patient suffering from a particular genetic disorder based on their medical record containing various information about the patient.</p>
<p>General Research Questions: What are the major conditions that help in early diagnosis of a genetic disorder in a patient? Are these factors inter-dependent on each other?</p>
<p>Dataset Description:</p>
<p>The genome and genetics dataset is collected by the National Center for Biotechnology Information (NCBI). The dataset includes patients with various genetic disorders, and the genetic disorders can be broadly categorized into three groups: mitochondrial genetic inheritance disorders, single-gene inheritance diseases, and multifactorial genetic inheritance disorders. These genetic disorders are just a few examples of the conditions that are included in the dataset.</p>
<p>Unit of Analysis: Each row represents classified medicial information like Age, Maternal and Paternal gene, Blood cell count, Respiratory and Heart rate, Test results, Previous abortions, Presence of symptoms, Birth defects and the Genetic disorder of a particular patient.</p>
<p>My study aims to analyse the relationship between different genetic factors, extracting insights based on correlation between medical precursors and test results, and predict the specific genetic disorder and its subclass for each patient based on their medical information.</p>
</section>
<section id="part-2.-background-of-the-topicliterature-review" class="level2">
<h2 class="anchored" data-anchor-id="part-2.-background-of-the-topicliterature-review">Part 2. Background of the Topic/Literature Review</h2>
<p>The exponential growth of the global population has resulted in a strain on essential resources such as healthcare, food, and shelter, which in turn has led to an increase in the incidence of genetic disorders. Genetic disorders are health conditions that typically result from DNA mutations or changes in the overall structure or number of chromosomes. Hereditary illnesses are becoming more prevalent due to insufficient awareness regarding the necessity of genetic testing. Tragically, such illnesses frequently result in the premature death of children, underscoring the vital significance of genetic testing during pregnancy.</p>
</section>
<section id="part-3.-dataset-introduction" class="level2">
<h2 class="anchored" data-anchor-id="part-3.-dataset-introduction">Part 3. Dataset Introduction</h2>
<p>The dataset includes patients with various genetic disorders, and the objective is to predict the specific genetic disorder and its subclass for each patient based on their medical information. Some of the disorder subclasses present in the dataset are Leigh syndrome, which is a rare inherited neurometabolic disorder, mitochondrial myopathy, which is a genetic disorder that affects the muscles and other organs that require a lot of energy, cystic fibrosis, which is a progressive genetic disorder that affects the lungs, pancreas, and other organs, Tay-Sachs, which is a rare inherited disorder that progressively destroys nerve cells in the brain and spinal cord, and diabetes, which is a metabolic disorder characterized by high blood sugar levels. These disorder subclasses are just a few examples of the conditions that are included in the dataset, and our project aims to study the subclass for each patient based on their medical information and analysing the correlation between them to help in pre-emptive detection of the disease.</p>
</section>
<section id="part-4.-dataset-description" class="level2">
<h2 class="anchored" data-anchor-id="part-4.-dataset-description">Part 4. Dataset Description</h2>
<p>(Story) This dataset contains a total of 22083 rows and 45 columns and the descriptive information about each of the 45 features are given below:</p>
<pre><code>-     Patient Id: Represents the unique identification number of a patient
-     Patient Age: Represents the age of a patient
-     Genes in mother's side: Represents a gene defect in the patient's mother
-     Inherited from father: Represents a gene defect in the patient's father
-     Maternal gene: Represents a gene defect in the patient's maternal side of the family
-     Paternal gene: Represents a gene defect in the patient's paternal side of the family
-     Blood cell count (mcL): Represents the blood cell count of a patient
-     Patient First Name: Represents a patient's first name
-     Family Name: Represents a patient's family name or surname
-     Father's name: Represents a patient's father's name
-    Mother's age: Represents a patient's mother's name                                    
-     Father's age: Represents a patient's father's age                                   
-     Institute Name: Represents the medical institute where a patient was born                   
-     Location of Institute: Represents the location of the medical institute                      
-     Status: Represents whether a patient is deceased                                         
-     Respiratory Rate (breaths/min): Represents a patient's respiratory breathing rate           
-    Heart Rate (rates/min): Represents a patient's heart rate                          
-     Test 1 to Test 5: Represents different (masked) tests that were conducted on a patient
-     Parental consent: Represents whether a patient's parents approved the treatment plan        
-    Follow-up: Represents a patient's level of risk (how intense their condition is)  
-     Gender: Represents a patient's gender                                         
-     Birth asphyxia: Represents whether a patient suffered from birth asphyxia                    
-     Autopsy shows birth defect (if applicable): Represents whether a patient's autopsy showed any birth defects     
-     Place of birth: Represents whether a patient was born in a medical institute or home        
-     Folic acid details (peri-conceptional): Represents the periconceptional folic acid supplementation details of a patient          
-    H/O serious maternal illness: Represents an unexpected outcome of labor and delivery that resulted in significant short or long-term consequences to a patient's mother                 
-    H/O radiation exposure (x-ray): Represents whether a patient has any radiation exposure history     
-    H/O substance abuse: Represents whether a parent has a history of drug addiction             
-    Assisted conception IVF/ART: Represents the type of treatment used for infertility            
-    History of anomalies in previous pregnancies: Represents whether the mother had any anomalies in her previous pregnancies   
-    No. of previous abortion: Represents the number of abortions that a mother had                
-    Birth defects: Represents whether a patient has birth defects                                
-    White Blood cell count (thousand per microliter): Represents a patient's white blood cell count
-    Blood test result: Represents a patient's blood test results                              
-    Symptom 1 to Symptom 5: Represents (masked) different types of symptoms that a patient had    
-    Genetic Disorder: Represents the genetic disorder that a patient has                          
-    Disorder Subclass: Represents the subclass of the disorder</code></pre>
<p>Coding Component:</p>
<ol type="1">
<li>Read the titanic dataset;</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read and get a overview of the data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">read_csv</span>(<span class="st">"/Users/avineshkrishnan/Desktop/FnuAvineshKrishnan_FinalProjectData/train_genetic_disorders.csv"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">view</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li><p>Descriptive information of the titanic dataset :</p>
<p>::: {.cell}</p>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Checking the dimensions of the data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>::: {.cell-output .cell-output-stdout} <code>[1] 22083    45</code> :::</p>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># checking the number of unique disorder sub classes present </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>::: {.cell-output .cell-output-stdout} <code>[1] 10</code> :::</p>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#printing out the first few examples</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>::: {.cell-output-display}</p>

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Patient Id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Patient Age"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Genes in mother's side"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Inherited from father"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Maternal gene"],"name":[5],"type":["chr"],"align":["left"]},{"label":["Paternal gene"],"name":[6],"type":["chr"],"align":["left"]},{"label":["Blood cell count (mcL)"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Patient First Name"],"name":[8],"type":["chr"],"align":["left"]},{"label":["Family Name"],"name":[9],"type":["chr"],"align":["left"]},{"label":["Father's name"],"name":[10],"type":["chr"],"align":["left"]},{"label":["Mother's age"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["Father's age"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["Institute Name"],"name":[13],"type":["chr"],"align":["left"]},{"label":["Location of Institute"],"name":[14],"type":["chr"],"align":["left"]},{"label":["Status"],"name":[15],"type":["chr"],"align":["left"]},{"label":["Respiratory Rate (breaths/min)"],"name":[16],"type":["chr"],"align":["left"]},{"label":["Heart Rate (rates/min"],"name":[17],"type":["chr"],"align":["left"]},{"label":["Test 1"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["Test 2"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["Test 3"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["Test 4"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["Test 5"],"name":[22],"type":["dbl"],"align":["right"]},{"label":["Parental consent"],"name":[23],"type":["chr"],"align":["left"]},{"label":["Follow-up"],"name":[24],"type":["chr"],"align":["left"]},{"label":["Gender"],"name":[25],"type":["chr"],"align":["left"]},{"label":["Birth asphyxia"],"name":[26],"type":["chr"],"align":["left"]},{"label":["Autopsy shows birth defect (if applicable)"],"name":[27],"type":["chr"],"align":["left"]},{"label":["Place of birth"],"name":[28],"type":["chr"],"align":["left"]},{"label":["Folic acid details (peri-conceptional)"],"name":[29],"type":["chr"],"align":["left"]},{"label":["H/O serious maternal illness"],"name":[30],"type":["chr"],"align":["left"]},{"label":["H/O radiation exposure (x-ray)"],"name":[31],"type":["chr"],"align":["left"]},{"label":["H/O substance abuse"],"name":[32],"type":["chr"],"align":["left"]},{"label":["Assisted conception IVF/ART"],"name":[33],"type":["chr"],"align":["left"]},{"label":["History of anomalies in previous pregnancies"],"name":[34],"type":["chr"],"align":["left"]},{"label":["No. of previous abortion"],"name":[35],"type":["dbl"],"align":["right"]},{"label":["Birth defects"],"name":[36],"type":["chr"],"align":["left"]},{"label":["White Blood cell count (thousand per microliter)"],"name":[37],"type":["dbl"],"align":["right"]},{"label":["Blood test result"],"name":[38],"type":["chr"],"align":["left"]},{"label":["Symptom 1"],"name":[39],"type":["dbl"],"align":["right"]},{"label":["Symptom 2"],"name":[40],"type":["dbl"],"align":["right"]},{"label":["Symptom 3"],"name":[41],"type":["dbl"],"align":["right"]},{"label":["Symptom 4"],"name":[42],"type":["dbl"],"align":["right"]},{"label":["Symptom 5"],"name":[43],"type":["dbl"],"align":["right"]},{"label":["Genetic Disorder"],"name":[44],"type":["chr"],"align":["left"]},{"label":["Disorder Subclass"],"name":[45],"type":["chr"],"align":["left"]}],"data":[{"1":"PID0x6418","2":"2","3":"Yes","4":"No","5":"Yes","6":"No","7":"4.760603","8":"Richard","9":"NA","10":"Larre","11":"NA","12":"NA","13":"Boston Specialty & Rehabilitation Hospital","14":"55 FRUIT ST\\nCENTRAL, MA 02114\\n(42.36247485742686, -71.06924724545246)","15":"Alive","16":"Normal (30-60)","17":"Normal","18":"0","19":"NA","20":"NA","21":"1","22":"0","23":"Yes","24":"High","25":"NA","26":"NA","27":"Not applicable","28":"Institute","29":"No","30":"NA","31":"No","32":"No","33":"No","34":"Yes","35":"NA","36":"NA","37":"9.857562","38":"NA","39":"1","40":"1","41":"1","42":"1","43":"1","44":"Mitochondrial genetic inheritance disorders","45":"Leber's hereditary optic neuropathy"},{"1":"PID0x25d5","2":"4","3":"Yes","4":"Yes","5":"No","6":"No","7":"4.910669","8":"Mike","9":"NA","10":"Brycen","11":"NA","12":"23","13":"St. Margaret's Hospital For Women","14":"1515 COMMONWEALTH AV\\nALLSTON/BRIGHTON, MA 02135\\n(42.34665771451756, -71.14136122385321)","15":"Deceased","16":"Tachypnea","17":"Normal","18":"NA","19":"0","20":"0","21":"1","22":"0","23":"Yes","24":"High","25":"NA","26":"No","27":"None","28":"NA","29":"Yes","30":"Yes","31":"Not applicable","32":"Not applicable","33":"No","34":"Yes","35":"NA","36":"Multiple","37":"5.522560","38":"normal","39":"1","40":"NA","41":"1","42":"1","43":"0","44":"NA","45":"Cystic fibrosis"},{"1":"PID0x4a82","2":"6","3":"Yes","4":"No","5":"No","6":"No","7":"4.893297","8":"Kimberly","9":"NA","10":"Nashon","11":"41","12":"22","13":"NA","14":"-","15":"Alive","16":"Normal (30-60)","17":"Tachycardia","18":"0","19":"0","20":"0","21":"1","22":"0","23":"Yes","24":"Low","25":"NA","26":"No record","27":"Not applicable","28":"NA","29":"Yes","30":"No","31":"Yes","32":"NA","33":"Yes","34":"Yes","35":"4","36":"Singular","37":"NA","38":"normal","39":"0","40":"1","41":"1","42":"1","43":"1","44":"Multifactorial genetic inheritance disorders","45":"Diabetes"},{"1":"PID0x4ac8","2":"12","3":"Yes","4":"No","5":"Yes","6":"No","7":"4.705280","8":"Jeffery","9":"Hoelscher","10":"Aayaan","11":"21","12":"NA","13":"NA","14":"55 FRUIT ST\\nCENTRAL, MA 02114\\n(42.36247485742686, -71.06924724545246)","15":"Deceased","16":"Tachypnea","17":"Normal","18":"0","19":"0","20":"0","21":"1","22":"0","23":"Yes","24":"High","25":"Male","26":"Not available","27":"No","28":"Institute","29":"No","30":"Yes","31":"-","32":"Not applicable","33":"NA","34":"Yes","35":"1","36":"Singular","37":"7.919321","38":"inconclusive","39":"0","40":"0","41":"1","42":"0","43":"0","44":"Mitochondrial genetic inheritance disorders","45":"Leigh syndrome"},{"1":"PID0x1bf7","2":"11","3":"Yes","4":"No","5":"NA","6":"Yes","7":"4.720703","8":"Johanna","9":"Stutzman","10":"Suave","11":"32","12":"NA","13":"Carney Hospital","14":"300 LONGWOOD AV\\nFENWAY/KENMORE, MA 02115\\n(42.337592548462226, -71.10472284437952)","15":"Alive","16":"Tachypnea","17":"Tachycardia","18":"0","19":"0","20":"0","21":"1","22":"0","23":"NA","24":"Low","25":"Male","26":"Not available","27":"Not applicable","28":"Institute","29":"No","30":"Yes","31":"-","32":"Not applicable","33":"Yes","34":"No","35":"4","36":"Multiple","37":"4.098210","38":"NA","39":"0","40":"0","41":"0","42":"0","43":"NA","44":"Multifactorial genetic inheritance disorders","45":"Cancer"},{"1":"PID0x44fe","2":"14","3":"Yes","4":"No","5":"Yes","6":"No","7":"5.103188","8":"Richard","9":"NA","10":"Coleston","11":"NA","12":"NA","13":"Massachusetts General Hospital","14":"55 FRUIT ST\\nCENTRAL, MA 02114\\n(42.36247485742686, -71.06924724545246)","15":"Deceased","16":"NA","17":"Normal","18":"0","19":"0","20":"0","21":"1","22":"0","23":"Yes","24":"Low","25":"Female","26":"Not available","27":"None","28":"Institute","29":"No","30":"No","31":"No","32":"No","33":"NA","34":"No","35":"0","36":"Multiple","37":"10.272230","38":"normal","39":"1","40":"0","41":"0","42":"1","43":"0","44":"Single-gene inheritance diseases","45":"Cystic fibrosis"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>::: :::</p></li>
<li><p>Conduct summary statistics of the dataset :</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying the summary of the Patients data in a table</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> <span class="fu">summary</span>(data)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(summary_stats)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Patient Id</th>
<th style="text-align: left;">Patient Age</th>
<th style="text-align: left;">Genes in mother’s side</th>
<th style="text-align: left;">Inherited from father</th>
<th style="text-align: left;">Maternal gene</th>
<th style="text-align: left;">Paternal gene</th>
<th style="text-align: left;">Blood cell count (mcL)</th>
<th style="text-align: left;">Patient First Name</th>
<th style="text-align: left;">Family Name</th>
<th style="text-align: left;">Father’s name</th>
<th style="text-align: left;">Mother’s age</th>
<th style="text-align: left;">Father’s age</th>
<th style="text-align: left;">Institute Name</th>
<th style="text-align: left;">Location of Institute</th>
<th style="text-align: left;">Status</th>
<th style="text-align: left;">Respiratory Rate (breaths/min)</th>
<th style="text-align: left;">Heart Rate (rates/min</th>
<th style="text-align: left;">Test 1</th>
<th style="text-align: left;">Test 2</th>
<th style="text-align: left;">Test 3</th>
<th style="text-align: left;">Test 4</th>
<th style="text-align: left;">Test 5</th>
<th style="text-align: left;">Parental consent</th>
<th style="text-align: left;">Follow-up</th>
<th style="text-align: left;">Gender</th>
<th style="text-align: left;">Birth asphyxia</th>
<th style="text-align: left;">Autopsy shows birth defect (if applicable)</th>
<th style="text-align: left;">Place of birth</th>
<th style="text-align: left;">Folic acid details (peri-conceptional)</th>
<th style="text-align: left;">H/O serious maternal illness</th>
<th style="text-align: left;">H/O radiation exposure (x-ray)</th>
<th style="text-align: left;">H/O substance abuse</th>
<th style="text-align: left;">Assisted conception IVF/ART</th>
<th style="text-align: left;">History of anomalies in previous pregnancies</th>
<th style="text-align: left;">No.&nbsp;of previous abortion</th>
<th style="text-align: left;">Birth defects</th>
<th style="text-align: left;">White Blood cell count (thousand per microliter)</th>
<th style="text-align: left;">Blood test result</th>
<th style="text-align: left;">Symptom 1</th>
<th style="text-align: left;">Symptom 2</th>
<th style="text-align: left;">Symptom 3</th>
<th style="text-align: left;">Symptom 4</th>
<th style="text-align: left;">Symptom 5</th>
<th style="text-align: left;">Genetic Disorder</th>
<th style="text-align: left;">Disorder Subclass</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. : 0.000</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. :4.093</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. :18.00</td>
<td style="text-align: left;">Min. :20.00</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. :0</td>
<td style="text-align: left;">Min. :0</td>
<td style="text-align: left;">Min. :0</td>
<td style="text-align: left;">Min. :1</td>
<td style="text-align: left;">Min. :0</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. :0</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. : 3.000</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Min. :0.000</td>
<td style="text-align: left;">Min. :0.000</td>
<td style="text-align: left;">Min. :0.0000</td>
<td style="text-align: left;">Min. :0.0000</td>
<td style="text-align: left;">Min. :0.0000</td>
<td style="text-align: left;">Length:22083</td>
<td style="text-align: left;">Length:22083</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.: 3.000</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.:4.763</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.:26.00</td>
<td style="text-align: left;">1st Qu.:31.00</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.:0</td>
<td style="text-align: left;">1st Qu.:0</td>
<td style="text-align: left;">1st Qu.:0</td>
<td style="text-align: left;">1st Qu.:1</td>
<td style="text-align: left;">1st Qu.:0</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.:1</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.: 5.419</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">1st Qu.:0.000</td>
<td style="text-align: left;">1st Qu.:0.000</td>
<td style="text-align: left;">1st Qu.:0.0000</td>
<td style="text-align: left;">1st Qu.:0.0000</td>
<td style="text-align: left;">1st Qu.:0.0000</td>
<td style="text-align: left;">Class :character</td>
<td style="text-align: left;">Class :character</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median : 7.000</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median :4.899</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median :35.00</td>
<td style="text-align: left;">Median :42.00</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median :0</td>
<td style="text-align: left;">Median :0</td>
<td style="text-align: left;">Median :0</td>
<td style="text-align: left;">Median :1</td>
<td style="text-align: left;">Median :0</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median :2</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median : 7.473</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Median :1.000</td>
<td style="text-align: left;">Median :1.000</td>
<td style="text-align: left;">Median :1.0000</td>
<td style="text-align: left;">Median :0.0000</td>
<td style="text-align: left;">Median :0.0000</td>
<td style="text-align: left;">Mode :character</td>
<td style="text-align: left;">Mode :character</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean : 6.975</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean :4.899</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean :34.52</td>
<td style="text-align: left;">Mean :41.94</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean :0</td>
<td style="text-align: left;">Mean :0</td>
<td style="text-align: left;">Mean :0</td>
<td style="text-align: left;">Mean :1</td>
<td style="text-align: left;">Mean :0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean :2</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean : 7.485</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Mean :0.592</td>
<td style="text-align: left;">Mean :0.553</td>
<td style="text-align: left;">Mean :0.5374</td>
<td style="text-align: left;">Mean :0.4974</td>
<td style="text-align: left;">Mean :0.4608</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:11.000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:5.034</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:43.00</td>
<td style="text-align: left;">3rd Qu.:53.00</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:0</td>
<td style="text-align: left;">3rd Qu.:0</td>
<td style="text-align: left;">3rd Qu.:0</td>
<td style="text-align: left;">3rd Qu.:1</td>
<td style="text-align: left;">3rd Qu.:0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:3</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.: 9.529</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">3rd Qu.:1.000</td>
<td style="text-align: left;">3rd Qu.:1.000</td>
<td style="text-align: left;">3rd Qu.:1.0000</td>
<td style="text-align: left;">3rd Qu.:1.0000</td>
<td style="text-align: left;">3rd Qu.:1.0000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :14.000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :5.610</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :51.00</td>
<td style="text-align: left;">Max. :64.00</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :0</td>
<td style="text-align: left;">Max. :0</td>
<td style="text-align: left;">Max. :0</td>
<td style="text-align: left;">Max. :1</td>
<td style="text-align: left;">Max. :0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :4</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :12.000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Max. :1.000</td>
<td style="text-align: left;">Max. :1.000</td>
<td style="text-align: left;">Max. :1.0000</td>
<td style="text-align: left;">Max. :1.0000</td>
<td style="text-align: left;">Max. :1.0000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :2440</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :1072</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :6790</td>
<td style="text-align: left;">NA’s :6761</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :3091</td>
<td style="text-align: left;">NA’s :3125</td>
<td style="text-align: left;">NA’s :3113</td>
<td style="text-align: left;">NA’s :3121</td>
<td style="text-align: left;">NA’s :3144</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :3126</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :3118</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA’s :3128</td>
<td style="text-align: left;">NA’s :3184</td>
<td style="text-align: left;">NA’s :3075</td>
<td style="text-align: left;">NA’s :3096</td>
<td style="text-align: left;">NA’s :3127</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying number of NAs in key variables Genetic Disorder and Disorder Subclass</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3121</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3140</code></pre>
</div>
</div>
<p>Number of NA values in Genetic Disorder and Disorder Subclass are 3121 and 3140 respectively. We remove these at the start of our Data Pre-processing section since we don’t want to have missing values in our important features.</p>
</section>
<section id="part-5.-data-pre-processing-and-analysis-plan" class="level2">
<h2 class="anchored" data-anchor-id="part-5.-data-pre-processing-and-analysis-plan">Part 5. Data Pre-processing and Analysis Plan</h2>
<p>Specific questions :</p>
<ol type="1">
<li><p>How are features causing the genetic disorder related with one another?</p></li>
<li><p>How much variability is present in the important factors affecting the type of disease?</p></li>
<li><p>What is the distibution of categories within each feature influencing the disease?</p></li>
<li><p>What are the most common types of Genetic disorders children suffer from?</p></li>
<li><p>What are the most important parts of the medical information of a patient that helps in early diagnosis of the genetic disorder?</p></li>
</ol>
<p>Pre-processing Plan:</p>
<ul>
<li><p>Dropping null values from our main disease columns, Genetic Disorder and Disorder Subclass</p></li>
<li><p>Dropping features that are not important and do not affect our key disease columns</p></li>
<li><p>Analyzing unique values present in each feature to replace noisy values and null values with suitable values</p></li>
</ul>
<p>Analysis Plan :</p>
<ul>
<li><p>To answer the research questions mentioned above, I plan to conduct a combination of data analysis and visualization techniques.</p></li>
<li><p>Bar Charts and Histograms: I will create bar charts and histograms to visualize the relationship between categorical and numerical variables (e.g., Respiratory Rate, Heart Rate, Follow-up, Folic acid details), and their frequencies. This will allow for a quick study of distribution of sub categories of different features.</p></li>
<li><p>Scatter Plot: Scatter plots are valuable for analyzing the relationship between numerical variables, such as Patient Age, Blood cell count and White Blood cell count. They enable us to observe the dispersion of data points and determine if there is any correlation or trend between the variables.</p></li>
<li><p>Bivariate Visualization: Bivariate visualizations such as bar charts or stacked bar plots, are useful for comparing two variables. They allow us to visually examine the relationship between two categorical variables and allow us to derive insights about the inter-dependency of these features.</p></li>
</ul>
</section>
<section id="part-6.-data-pre-processing-and-visualization" class="level2">
<h2 class="anchored" data-anchor-id="part-6.-data-pre-processing-and-visualization">Part 6. Data Pre-processing and Visualization</h2>
<p>Pre-processing plan: Task 1 and 2</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping patients with missing values in 'Genetic Disorder' and 'Disorder Subclass' columns</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span><span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>My goal is to study the correlation of other features with these two disorder columns and NA values in these target variables would make that particular patient record useless. Hence I am removing rows which have null values in these target columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping specific columns from the dataset</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">select</span>(data, <span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Patient Id</span><span class="st">`</span>, <span class="st">`</span><span class="at">Family Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Institute Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Patient First Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Father's name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Place of birth</span><span class="st">`</span>, <span class="st">`</span><span class="at">Location of Institute</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Dropping features that do not affect the type of genetic disorder or subclass in any way and hence analyzing them isn’t of any use.</p>
<p>Pre-processing plan: Task 3 and Specific Questions 1, 2 and 3</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Checking unique values in 'Test 1', 'Test 2', 'Test 3', 'Test 4', and 'Test 5' columns</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  0 NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA  0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA  0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  1 NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  0 NA</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with -99.0 in 'Test 1', 'Test 2', 'Test 3', 'Test 4', and 'Test 5' columns</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Checking the unique values in the test results columns and replacing the missing values with an arbitrary negative value to identify them during future analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `Patient Age`</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"Patient age"</span>)) <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Patient Age"</span>) <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"red"</span>, <span class="at">labels =</span> <span class="st">"Patient age"</span>) <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `Blood cell count (mcL)` </span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"Blood cell count (mcL)"</span>)) <span class="sc">+</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"blue"</span>, <span class="at">labels =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `White Blood cell count (thousand per microliter)` </span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>)) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of White Blood cell count (thousand per microliter)"</span>) <span class="sc">+</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"black"</span>, <span class="at">labels =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>) <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I can see that Patient age is uniformly distributed from 0-14 verifying the summary statistic that our dataset includes children in that particular age range only. Blood cell count has a normal distribution around 4.9mcL showing that most children have their blood cell count between 4mcL and 5.5mcL with majority of the children having the mean blood cell count of about 4.9mcL. I also observed that the White blood cell count is somewhat uniformly distributed between 3 and 12 thousand per microliter telling us all children have varying White blood cell count which reflects in their varying strength of immune system.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Blood cell count (mcL)` against `White Blood cell count (thousand per #microliter)`</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"White Blood cell count vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the scatter plot, it looks like there are lot of outliers for the white blood cells. I can see that the blood cell count and white blood cell count do not have a strong positive or negative correlation, therefore the symptoms of in the blood sample are probably not caused by the white blood cells(leukocytes).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create scatter plot of `Blood cell count (mcL)` against `Patient Age`</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient Age"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Patient Age vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create scatter plot of `White Blood cell count (thousand per microliter)` against `Patient Age`</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient Age"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Patient Age vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the scatter plots, I can see that the patients are uniformly distributed over the range of blood cell counts and white blood cell counts giving us the idea that there is very little to no correlation between the Cell count and Age of the patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean values of `White Blood cell count (thousand per microliter)`,`Blood cell count #(mcL)`  and `Patient Age`</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>mean_wbc <span class="ot">&lt;-</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>mean_bc <span class="ot">&lt;-</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>mean_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean values</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>), mean_wbc, data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>), mean_bc, data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>), mean_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>As the cell count and age are important features, I am replacing the missing cell counts with the mean value and the missing age with floor of the mean value since age needs to be a round number.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Mother's age` against `Father's age`</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Father's age vs. Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I wanted to see if there is any correlation between the parents age which is clearly not the case from the views of the scatter plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot with fitted normal distribution for Mother's age</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">sd =</span> <span class="fu">sd</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Mother's age"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot with fitted normal distribution for Father's age</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">sd =</span> <span class="fu">sd</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Father's age"</span>) <span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Both the parents age follows a normal distribution being symmetric about their mean value. This shows that majority of fathers are about 43 years old and majority of the mothers are about 35 years old.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean values</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>mean_mother_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>mean_father_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean values</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>), mean_mother_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>), mean_father_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I am replacing the missing values in Father’s and Mother’s age with the floor of the mean value since age needs to be a round number.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot of `No. of previous abortion`</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">y =</span> ..density..)) <span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of No. of previous abortion"</span>) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Checking unique values in `No. of previous abortion`</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA  4  1  0  3  2</code></pre>
</div>
</div>
<p>I wanted to learn about the <code>No. of previous abortion</code> variable and from its density distribution, it is surprising to see that there are many children who have had 1,2,3 and even 4 abortions already. I want to check their Mothers age to get more insights.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Mother's age` against `No. of previous abortion`</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Mother's age vs. No. of previous abortion"</span>) <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Wow, there are cases where patients with just 18 year old mothers have had 4 abortions. I want to look into the Father’s age of these patients.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering data for Mother's age equal to 18</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>filtered_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(data, <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span> <span class="sc">==</span> <span class="dv">18</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating count Bar graph of Father's age for these records</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Father's age for 18 year old mother"</span>) <span class="sc">+</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I can see that majority of fathers of these patients who have a 18 year old mother and have had a few abortions are about 41 years old.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean value of `No. of previous abortion`</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>mean_previous_abortion <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean value</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>), mean_previous_abortion, data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I am replacing the missing values in No.&nbsp;of previous abortion with the floor of the mean value since the variable needs to be a round number.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values of Symptoms columns with 0.0</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span>)</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I am replacing the missing values in the Symptoms columns with 0 since 0 represents no symptoms and 1 represents that symptoms are present and when the symptoms aren’t mentioned, it is safe to assume that they have no symptoms.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating count Bar graphs for Categorical features</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Maternal gene"</span>) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Respiratory Rate (breaths/min)"</span>) <span class="sc">+</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"green"</span>) <span class="sc">+</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Heart Rate (rates/min)"</span>) <span class="sc">+</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Parental consent</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"orange"</span>) <span class="sc">+</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Parental consent"</span>) <span class="sc">+</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"purple"</span>) <span class="sc">+</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Follow-up"</span>) <span class="sc">+</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"pink"</span>) <span class="sc">+</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Folic acid details (peri-conceptional)"</span>) <span class="sc">+</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"brown"</span>) <span class="sc">+</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: H/O radiation exposure (x-ray)"</span>) <span class="sc">+</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: H/O substance abuse"</span>) <span class="sc">+</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"cyan"</span>) <span class="sc">+</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Assisted conception IVF/ART"</span>) <span class="sc">+</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"yellow"</span>) <span class="sc">+</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: History of anomalies in previous pregnancies"</span>) <span class="sc">+</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the Bar graphs of the counts of these categorical features, I can see which sub category of each variable accounts for the majority percentage. Since there are lot of parental consent, I will assume that tests conducted due to parents consent and there’s no way a kid will check up on his own.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping Parental consent column</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">select</span>(data, <span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Parental consent</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Dropping the parental consent since most of them require it and the feature does not have influence on the type of genetic disorder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Checking unique values in columns</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Yes" "No"  NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Normal (30-60)" "Tachypnea"      NA              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Normal"      "Tachycardia" NA           </code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "High" "Low"  NA    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"  "Yes" NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"             "Yes"            "-"              "Not applicable"
[5] NA              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"             NA               "Not applicable" "-"             
[5] "Yes"           </code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"  "Yes" NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Yes" "No"  NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA              "No record"     "Not available" "Yes"          
[5] "No"           </code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Gender</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA          "Male"      "Female"    "Ambiguous"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA         "Singular" "Multiple"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA                  "normal"            "inconclusive"     
[4] "slightly abnormal" "abnormal"         </code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA    "No"  "Yes"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"  "Yes" NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Not applicable" "No"             "None"           "Yes"           
[5] NA              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span>Status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Alive"    "Deceased"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Yes" "No" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No"  "Yes"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA with "No" in specific columns</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>)</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>)</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>)</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>)</span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>)</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>)</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>)</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>)</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>)</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>)</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate mode</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>Mode <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>  ux <span class="ot">&lt;-</span> <span class="fu">unique</span>(x)</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>  ux[<span class="fu">which.max</span>(<span class="fu">tabulate</span>(<span class="fu">match</span>(x, ux)))]</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA with the most frequent value in specific columns</span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>)</span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Gender <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span>Gender), <span class="fu">Mode</span>(data<span class="sc">$</span>Gender), data<span class="sc">$</span>Gender)</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>)</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>)</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span>Status), <span class="fu">Mode</span>(data<span class="sc">$</span>Status), data<span class="sc">$</span>Status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I checked the values in these columns and replaced the missing values in the categorical features with “No” since that would be most apt if the illness wasn’t mentioned and with value of the most frequent value(mode) in the numerical features since these variables have a partial but not so strong impact on the genetic disorder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing -99 with 0 in 'Test 1', 'Test 2', 'Test 3', 'Test 4', 'Test 5'</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing -99 in 'White Blood cell count (thousand per microliter)' with median</span></span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing '-99' with mode in 'Respiratory Rate (breaths/min)', 'Heart Rate (rates/min)'</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-99'</span>] <span class="ot">&lt;-</span> <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-99'</span>] <span class="ot">&lt;-</span> <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'Normal (30-60)' with 'Normal' in 'Respiratory Rate (breaths/min)'</span></span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'Normal (30-60)'</span>] <span class="ot">&lt;-</span> <span class="st">'Normal'</span></span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'No record' with 'Not available' in 'Birth asphyxia'</span></span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'No record'</span>] <span class="ot">&lt;-</span> <span class="st">'Not available'</span></span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'None' and '-' with 'No' in 'Autopsy shows birth defect (if applicable)', 'H/O radiation exposure (x-ray)', 'H/O substance abuse'</span></span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I replaced arbitrary negative value(-99) in Test results feature with 0 since it implies that missing value equals a negative test which is the most appropriate to assume. I replaced the arbitrary value in White Blood cell count, Respiratory and heart rate with median and mode respectively as taking the central tendency value would make the most sense as these features are key for determining the type of genetic disorder. I replaced ‘Normal (30-60)’ with ‘Normal’, ‘No record’ with ‘Not available’, ‘None’ and ‘-’ with ‘No’ in Respiratory Rate, Birth asphyxia, Autopsy shows birth defect, H/O radiation exposure and H/O substance abuse respectively to maintain uniformity within the sub categories.</p>
<p>Specific Question 4</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the count plot for `Disorder Subclass`</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Disorder Subclass"</span>, <span class="at">x =</span> <span class="st">"Disorder Subclass"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the Count plot for the subclasses of genetic disorder, I can say that most of the patients suffer from Leigh Syndrome while the least amount of children have Cancer.</p>
<p>Specific Question 5</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Maternal gene and Genetic Disorder</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Maternal gene"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Maternal gene vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Paternal gene and Genetic Disorder</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Paternal gene"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Paternal gene vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I observe that the maternal gene equally affects patients and is does not determine the type of genetic disorder whereas the absence of the paternal gene is seen in more cases of genetic abnormalities as compared to the patients containing the paternal gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Respiratory Rate (breaths/min) and Genetic Disorder</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Respiratory Rate (breaths/min)"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Respiratory Rate (breaths/min) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Heart Rate (rates/min and Genetic Disorder</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Heart Rate (rates/min"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Heart Rate (rates/min) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Birth asphyxia and Genetic Disorder</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Birth asphyxia"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Birth asphyxia vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I see that patients with a normal Respiratory breathing rate are in larger proportion when it comes to having a genetic disorder and hence I think that Respiratory rate may not be a primary factor for determining presence of a genetic disease. In the same manner, Heart rate and Birth asphyxia can also be taken out of consideration for early prognosis of a genetic disorder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between H/O radiation exposure (x-ray) and Genetic Disorder</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"H/O radiation exposure (x-ray)"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"H/O radiation exposure (x-ray) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I observe that people with no radiation exposure also suffer from a lot of genetic disorders compared to people exposed to x-rays, and this condition may also not be a part of predicting the presence of a genetic disease in the patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between No. of previous abortion and Genetic Disorder</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"No. of previous abortion vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the correlation plot, patients with even a single previous abortion tend to have a higher chance of suffering from a genetic disorder, mainly Mitochondrial Inheritance disorders.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Blood cell count (mcL) and Genetic Disorder</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">x =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>, <span class="at">x =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Blood cell count (mcL) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FnuAvineshKrishnan_final_project_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the graph, I see that patients with blood cell count within the full range suffer from genetic disorders which means that this feature may not be a key influence for causing genetic diseases in children.</p>
</section>
<section id="part-7.-conclusion-and-discussion" class="level2">
<h2 class="anchored" data-anchor-id="part-7.-conclusion-and-discussion">Part 7. Conclusion and Discussion :</h2>
<ul>
<li><p>The analysis of the dataset has provided valuable insights into the relationship between various features and the presence of genetic disorders. The first step was to remove rows with missing values in the target columns, as these records would not contribute to the analysis. Additionally, irrelevant features that did not affect the type of genetic disorder were dropped, streamlining the dataset for further examination. The examination of the test results columns revealed unique values and the replacement of missing values with an arbitrary negative value for future identification. The analysis of patient age, blood cell count, and white blood cell count showcased interesting patterns. Patient age was found to be uniformly distributed within the range of 0 to 14 years, indicating that the dataset primarily consists of children in this age group. Blood cell count exhibited a normal distribution, with the majority of children having counts between 4mcL and 5.5mcL. Similarly, white blood cell count showed a somewhat uniform distribution between 3 and 12 thousand per microliter, reflecting the varying strength of the immune system among children.</p></li>
<li><p>Scatter plots were employed to explore potential correlations between variables. The scatter plot of blood cell count and white blood cell count indicated no strong positive or negative correlation, suggesting that the symptoms observed in the blood sample may not be caused by variations in white blood cell count. Additionally, scatter plots of blood cell count versus patient age and parental age exhibited no discernible correlation, indicating that these factors may have limited influence on the occurrence of genetic disorders.</p></li>
<li><p>The examination of parental age distributions revealed normal distributions for both fathers and mothers, with the majority of fathers being around 43 years old and mothers around 35 years old. Missing values in father’s and mother’s age were replaced with the floor of the mean value to maintain consistency and roundness in the age representation. Exploring the “No.&nbsp;of previous abortion” variable provided insights into the frequency of abortions among patients. It was observed that there were cases where patients with 18-year-old mothers had undergone multiple abortions. Furthermore, the examination of father’s age in these cases revealed that the majority of fathers were around 41 years old.</p></li>
<li><p>Count plots were used to analyze categorical features, such as parental consent, and revealed the predominance of specific categories. As a result, the parental consent feature was dropped, as it was determined to have little influence on the type of genetic disorder. Missing values in categorical features were replaced with “No” to represent the absence of the mentioned condition. Furthermore, specific values such as “Normal (30-60)” were replaced with simplified representations like “Normal” for consistency.</p></li>
<li><p>Count plots of the subclasses of genetic disorders highlighted the distribution of disorders within the dataset. Leigh Syndrome was found to be the most prevalent disorder, while Cancer accounted for the fewest cases. The analysis of the maternal gene revealed that it did not significantly influence the type of genetic disorder, whereas the absence of the paternal gene was more prevalent in cases of genetic abnormalities.</p></li>
<li><p>The examination of features such as respiratory rate, heart rate, radiation exposure, and birth asphyxia indicated that these factors may not be primary indicators of genetic disorders. Similarly, the correlation plot showed that patients with previous abortions tended to have a higher likelihood of suffering from genetic disorders, particularly those related to mitochondrial inheritance. Lastly, the scatter plot of blood cell count versus genetic disorders suggested that blood cell count may not be a key determining factor in the presence of genetic diseases.</p></li>
<li><p>In conclusion, the analysis of the dataset provided valuable insights into the relationship between various features and the occurrence of genetic disorders. The examination of unique values, data distributions, scatter plots, and count plots enabled a deeper understanding of the dataset and facilitated the identification of influential factors. These findings contribute to the field of genetic disorder research and can potentially aid in early prognosis and understanding the underlying mechanisms of these disorders</p></li>
</ul>
</section>
<section id="part-8.-bibliography" class="level2">
<h2 class="anchored" data-anchor-id="part-8.-bibliography">Part 8. Bibliography :</h2>
<p>Dataset : https://www.kaggle.com/datasets/aibuzz/predict-the-genetic-disorders-datasetof-genomes</p>
<p>Programming Language : R</p>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="DACSS/601_Spring_2023" data-repo-id="R_kgDOI4JcJA" data-category="Announcements" data-category-id="" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb104" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Final Project: Fnu Avinesh Krishnan"</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Fnu Avinesh Krishnan"</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> "Final Project writeup"</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "05/22/2023"</span></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a><span class="co">    css: styles.css</span></span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span></span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a><span class="co">  - final_Project</span></span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: setup</span></span>
<span id="cb104-24"><a href="#cb104-24" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb104-25"><a href="#cb104-25" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb104-26"><a href="#cb104-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-27"><a href="#cb104-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb104-28"><a href="#cb104-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb104-29"><a href="#cb104-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-30"><a href="#cb104-30" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">message=</span><span class="cn">FALSE</span>)</span>
<span id="cb104-31"><a href="#cb104-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-32"><a href="#cb104-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-33"><a href="#cb104-33" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 1. Introduction </span></span>
<span id="cb104-34"><a href="#cb104-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-35"><a href="#cb104-35" aria-hidden="true" tabindex="-1"></a>Project Topic: The project aims to analyze factors of a patient suffering from a particular genetic disorder based on their medical record containing various information about the patient.</span>
<span id="cb104-36"><a href="#cb104-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-37"><a href="#cb104-37" aria-hidden="true" tabindex="-1"></a>General Research Questions: What are the major conditions that help in early diagnosis of a genetic disorder in a patient? Are these factors inter-dependent on each other?</span>
<span id="cb104-38"><a href="#cb104-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-39"><a href="#cb104-39" aria-hidden="true" tabindex="-1"></a>Dataset Description: </span>
<span id="cb104-40"><a href="#cb104-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-41"><a href="#cb104-41" aria-hidden="true" tabindex="-1"></a>The genome and genetics dataset is collected by the National Center for Biotechnology Information (NCBI). The dataset includes patients with various genetic disorders, and the genetic disorders can be broadly categorized into three groups: mitochondrial genetic inheritance disorders, single-gene inheritance diseases, and multifactorial genetic inheritance disorders. These genetic disorders are just a few examples of the conditions that are included in the dataset.</span>
<span id="cb104-42"><a href="#cb104-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-43"><a href="#cb104-43" aria-hidden="true" tabindex="-1"></a>Unit of Analysis: Each row represents classified medicial information like Age, Maternal and Paternal gene, Blood cell count, Respiratory and Heart rate, Test results, Previous abortions, Presence of symptoms, Birth defects and the Genetic disorder of a particular patient.</span>
<span id="cb104-44"><a href="#cb104-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-45"><a href="#cb104-45" aria-hidden="true" tabindex="-1"></a>My study aims to analyse the relationship between different genetic factors, extracting insights based on correlation between medical precursors and test results, and predict the specific genetic disorder and its subclass for each patient based on their medical information.</span>
<span id="cb104-46"><a href="#cb104-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-47"><a href="#cb104-47" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 2. Background of the Topic/Literature Review</span></span>
<span id="cb104-48"><a href="#cb104-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-49"><a href="#cb104-49" aria-hidden="true" tabindex="-1"></a>The exponential growth of the global population has resulted in a strain on essential resources such as healthcare, food, and shelter, which in turn has led to an increase in the incidence of genetic disorders. Genetic disorders are health conditions that typically result from DNA mutations or changes in the overall structure or number of chromosomes. Hereditary illnesses are becoming more prevalent due to insufficient awareness regarding the necessity of genetic testing. Tragically, such illnesses frequently result in the premature death of children, underscoring the vital significance of genetic testing during pregnancy.</span>
<span id="cb104-50"><a href="#cb104-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-51"><a href="#cb104-51" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 3. Dataset Introduction</span></span>
<span id="cb104-52"><a href="#cb104-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-53"><a href="#cb104-53" aria-hidden="true" tabindex="-1"></a>The dataset includes patients with various genetic disorders, and the objective is to predict the specific genetic disorder and its subclass for each patient based on their medical information. Some of the disorder subclasses present in the dataset are Leigh syndrome, which is a rare inherited neurometabolic disorder, mitochondrial myopathy, which is a genetic disorder that affects the muscles and other organs that require a lot of energy, cystic fibrosis, which is a progressive genetic disorder that affects the lungs, pancreas, and other organs, Tay-Sachs, which is a rare inherited disorder that progressively destroys nerve cells in the brain and spinal cord, and diabetes, which is a metabolic disorder characterized by high blood sugar levels. These disorder subclasses are just a few examples of the conditions that are included in the dataset, and our project aims to study the subclass for each patient based on their medical information and analysing the correlation between them to help in pre-emptive detection of the disease. </span>
<span id="cb104-54"><a href="#cb104-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-55"><a href="#cb104-55" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 4. Dataset Description</span></span>
<span id="cb104-56"><a href="#cb104-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-57"><a href="#cb104-57" aria-hidden="true" tabindex="-1"></a>(Story)</span>
<span id="cb104-58"><a href="#cb104-58" aria-hidden="true" tabindex="-1"></a>This dataset contains a total of 22083 rows and 45 columns and the descriptive information about each of the 45 features are given below: </span>
<span id="cb104-59"><a href="#cb104-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-60"><a href="#cb104-60" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Patient Id: Represents the unique identification number of a patient</span></span>
<span id="cb104-61"><a href="#cb104-61" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Patient Age: Represents the age of a patient</span></span>
<span id="cb104-62"><a href="#cb104-62" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Genes in mother's side: Represents a gene defect in the patient's mother</span></span>
<span id="cb104-63"><a href="#cb104-63" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Inherited from father: Represents a gene defect in the patient's father</span></span>
<span id="cb104-64"><a href="#cb104-64" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Maternal gene: Represents a gene defect in the patient's maternal side of the family</span></span>
<span id="cb104-65"><a href="#cb104-65" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Paternal gene: Represents a gene defect in the patient's paternal side of the family</span></span>
<span id="cb104-66"><a href="#cb104-66" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Blood cell count (mcL): Represents the blood cell count of a patient</span></span>
<span id="cb104-67"><a href="#cb104-67" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Patient First Name: Represents a patient's first name</span></span>
<span id="cb104-68"><a href="#cb104-68" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Family Name: Represents a patient's family name or surname</span></span>
<span id="cb104-69"><a href="#cb104-69" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Father's name: Represents a patient's father's name</span></span>
<span id="cb104-70"><a href="#cb104-70" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Mother's age: Represents a patient's mother's name                                    </span></span>
<span id="cb104-71"><a href="#cb104-71" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Father's age: Represents a patient's father's age                                   </span></span>
<span id="cb104-72"><a href="#cb104-72" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Institute Name: Represents the medical institute where a patient was born                   </span></span>
<span id="cb104-73"><a href="#cb104-73" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Location of Institute: Represents the location of the medical institute                      </span></span>
<span id="cb104-74"><a href="#cb104-74" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Status: Represents whether a patient is deceased                                         </span></span>
<span id="cb104-75"><a href="#cb104-75" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Respiratory Rate (breaths/min): Represents a patient's respiratory breathing rate           </span></span>
<span id="cb104-76"><a href="#cb104-76" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Heart Rate (rates/min): Represents a patient's heart rate                          </span></span>
<span id="cb104-77"><a href="#cb104-77" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Test 1 to Test 5: Represents different (masked) tests that were conducted on a patient</span></span>
<span id="cb104-78"><a href="#cb104-78" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Parental consent: Represents whether a patient's parents approved the treatment plan        </span></span>
<span id="cb104-79"><a href="#cb104-79" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Follow-up: Represents a patient's level of risk (how intense their condition is)  </span></span>
<span id="cb104-80"><a href="#cb104-80" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Gender: Represents a patient's gender                                         </span></span>
<span id="cb104-81"><a href="#cb104-81" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Birth asphyxia: Represents whether a patient suffered from birth asphyxia                    </span></span>
<span id="cb104-82"><a href="#cb104-82" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Autopsy shows birth defect (if applicable): Represents whether a patient's autopsy showed any birth defects     </span></span>
<span id="cb104-83"><a href="#cb104-83" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Place of birth: Represents whether a patient was born in a medical institute or home        </span></span>
<span id="cb104-84"><a href="#cb104-84" aria-hidden="true" tabindex="-1"></a><span class="in">    -     Folic acid details (peri-conceptional): Represents the periconceptional folic acid supplementation details of a patient          </span></span>
<span id="cb104-85"><a href="#cb104-85" aria-hidden="true" tabindex="-1"></a><span class="in">    -    H/O serious maternal illness: Represents an unexpected outcome of labor and delivery that resulted in significant short or long-term consequences to a patient's mother                 </span></span>
<span id="cb104-86"><a href="#cb104-86" aria-hidden="true" tabindex="-1"></a><span class="in">    -    H/O radiation exposure (x-ray): Represents whether a patient has any radiation exposure history     </span></span>
<span id="cb104-87"><a href="#cb104-87" aria-hidden="true" tabindex="-1"></a><span class="in">    -    H/O substance abuse: Represents whether a parent has a history of drug addiction             </span></span>
<span id="cb104-88"><a href="#cb104-88" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Assisted conception IVF/ART: Represents the type of treatment used for infertility            </span></span>
<span id="cb104-89"><a href="#cb104-89" aria-hidden="true" tabindex="-1"></a><span class="in">    -    History of anomalies in previous pregnancies: Represents whether the mother had any anomalies in her previous pregnancies   </span></span>
<span id="cb104-90"><a href="#cb104-90" aria-hidden="true" tabindex="-1"></a><span class="in">    -    No. of previous abortion: Represents the number of abortions that a mother had                </span></span>
<span id="cb104-91"><a href="#cb104-91" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Birth defects: Represents whether a patient has birth defects                                </span></span>
<span id="cb104-92"><a href="#cb104-92" aria-hidden="true" tabindex="-1"></a><span class="in">    -    White Blood cell count (thousand per microliter): Represents a patient's white blood cell count</span></span>
<span id="cb104-93"><a href="#cb104-93" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Blood test result: Represents a patient's blood test results                              </span></span>
<span id="cb104-94"><a href="#cb104-94" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Symptom 1 to Symptom 5: Represents (masked) different types of symptoms that a patient had    </span></span>
<span id="cb104-95"><a href="#cb104-95" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Genetic Disorder: Represents the genetic disorder that a patient has                          </span></span>
<span id="cb104-96"><a href="#cb104-96" aria-hidden="true" tabindex="-1"></a><span class="in">    -    Disorder Subclass: Represents the subclass of the disorder</span></span>
<span id="cb104-97"><a href="#cb104-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-98"><a href="#cb104-98" aria-hidden="true" tabindex="-1"></a>Coding Component:</span>
<span id="cb104-99"><a href="#cb104-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-100"><a href="#cb104-100" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Read the titanic dataset;</span>
<span id="cb104-101"><a href="#cb104-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-104"><a href="#cb104-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-105"><a href="#cb104-105" aria-hidden="true" tabindex="-1"></a><span class="co">#read and get a overview of the data</span></span>
<span id="cb104-106"><a href="#cb104-106" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb104-107"><a href="#cb104-107" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">read_csv</span>(<span class="st">"/Users/avineshkrishnan/Desktop/FnuAvineshKrishnan_FinalProjectData/train_genetic_disorders.csv"</span>)</span>
<span id="cb104-108"><a href="#cb104-108" aria-hidden="true" tabindex="-1"></a><span class="fu">view</span>(data)</span>
<span id="cb104-109"><a href="#cb104-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-110"><a href="#cb104-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-111"><a href="#cb104-111" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Descriptive information of the titanic dataset :</span>
<span id="cb104-112"><a href="#cb104-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-113"><a href="#cb104-113" aria-hidden="true" tabindex="-1"></a>    <span class="in">```{r}</span></span>
<span id="cb104-114"><a href="#cb104-114" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Checking the dimensions of the data</span></span>
<span id="cb104-115"><a href="#cb104-115" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dim</span>(data)</span>
<span id="cb104-116"><a href="#cb104-116" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb104-117"><a href="#cb104-117" aria-hidden="true" tabindex="-1"></a>    <span class="co"># checking the number of unique disorder sub classes present </span></span>
<span id="cb104-118"><a href="#cb104-118" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(<span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>))</span>
<span id="cb104-119"><a href="#cb104-119" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb104-120"><a href="#cb104-120" aria-hidden="true" tabindex="-1"></a>    <span class="co">#printing out the first few examples</span></span>
<span id="cb104-121"><a href="#cb104-121" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(data)</span>
<span id="cb104-122"><a href="#cb104-122" aria-hidden="true" tabindex="-1"></a>    <span class="in">```</span></span>
<span id="cb104-123"><a href="#cb104-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-124"><a href="#cb104-124" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>Conduct summary statistics of the dataset : </span>
<span id="cb104-125"><a href="#cb104-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-128"><a href="#cb104-128" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-129"><a href="#cb104-129" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb104-130"><a href="#cb104-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-131"><a href="#cb104-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying the summary of the Patients data in a table</span></span>
<span id="cb104-132"><a href="#cb104-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-133"><a href="#cb104-133" aria-hidden="true" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> <span class="fu">summary</span>(data)</span>
<span id="cb104-134"><a href="#cb104-134" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(summary_stats)</span>
<span id="cb104-135"><a href="#cb104-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-136"><a href="#cb104-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-139"><a href="#cb104-139" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-140"><a href="#cb104-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying number of NAs in key variables Genetic Disorder and Disorder Subclass</span></span>
<span id="cb104-141"><a href="#cb104-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-142"><a href="#cb104-142" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>))</span>
<span id="cb104-143"><a href="#cb104-143" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>))</span>
<span id="cb104-144"><a href="#cb104-144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-145"><a href="#cb104-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-146"><a href="#cb104-146" aria-hidden="true" tabindex="-1"></a>Number of NA values in Genetic Disorder and Disorder Subclass are 3121 and 3140 respectively. We remove these at the start of our Data Pre-processing section since we don't want to have missing values in our important features.</span>
<span id="cb104-147"><a href="#cb104-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-148"><a href="#cb104-148" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 5. Data Pre-processing and Analysis Plan</span></span>
<span id="cb104-149"><a href="#cb104-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-150"><a href="#cb104-150" aria-hidden="true" tabindex="-1"></a>Specific questions : </span>
<span id="cb104-151"><a href="#cb104-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-152"><a href="#cb104-152" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>How are features causing the genetic disorder related with one another?</span>
<span id="cb104-153"><a href="#cb104-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-154"><a href="#cb104-154" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>How much variability is present in the important factors affecting the type of disease?</span>
<span id="cb104-155"><a href="#cb104-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-156"><a href="#cb104-156" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>What is the distibution of categories within each feature influencing the disease?</span>
<span id="cb104-157"><a href="#cb104-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-158"><a href="#cb104-158" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>What are the most common types of Genetic disorders children suffer from?</span>
<span id="cb104-159"><a href="#cb104-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-160"><a href="#cb104-160" aria-hidden="true" tabindex="-1"></a><span class="ss">5. </span>What are the most important parts of the medical information of a patient that helps in early diagnosis of the genetic disorder?</span>
<span id="cb104-161"><a href="#cb104-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-162"><a href="#cb104-162" aria-hidden="true" tabindex="-1"></a>Pre-processing Plan:</span>
<span id="cb104-163"><a href="#cb104-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-164"><a href="#cb104-164" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Dropping null values from our main disease columns, Genetic Disorder and Disorder Subclass</span>
<span id="cb104-165"><a href="#cb104-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-166"><a href="#cb104-166" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Dropping features that are not important and do not affect our key disease columns</span>
<span id="cb104-167"><a href="#cb104-167" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb104-168"><a href="#cb104-168" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Analyzing unique values present in each feature to replace noisy values and null values with suitable values</span>
<span id="cb104-169"><a href="#cb104-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-170"><a href="#cb104-170" aria-hidden="true" tabindex="-1"></a>Analysis Plan :</span>
<span id="cb104-171"><a href="#cb104-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-172"><a href="#cb104-172" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>To answer the research questions mentioned above, I plan to conduct a combination of data analysis and visualization techniques.</span>
<span id="cb104-173"><a href="#cb104-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-174"><a href="#cb104-174" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Bar Charts and Histograms: I will create bar charts and histograms to visualize the relationship between categorical and numerical variables (e.g., Respiratory Rate, Heart Rate, Follow-up, Folic acid details), and their frequencies. This will allow for a quick study of distribution of sub categories of different features.</span>
<span id="cb104-175"><a href="#cb104-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-176"><a href="#cb104-176" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Scatter Plot: Scatter plots are valuable for analyzing the relationship between numerical variables, such as Patient Age, Blood cell count and White Blood cell count. They enable us to observe the dispersion of data points and determine if there is any correlation or trend between the variables.</span>
<span id="cb104-177"><a href="#cb104-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-178"><a href="#cb104-178" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Bivariate Visualization: Bivariate visualizations such as bar charts or stacked bar plots, are useful for comparing two variables. They allow us to visually examine the relationship between two categorical variables and allow us to derive insights about the inter-dependency of these features.</span>
<span id="cb104-179"><a href="#cb104-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-180"><a href="#cb104-180" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 6. Data Pre-processing and Visualization</span></span>
<span id="cb104-181"><a href="#cb104-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-182"><a href="#cb104-182" aria-hidden="true" tabindex="-1"></a>Pre-processing plan: Task 1 and 2</span>
<span id="cb104-183"><a href="#cb104-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-186"><a href="#cb104-186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-187"><a href="#cb104-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping patients with missing values in 'Genetic Disorder' and 'Disorder Subclass' columns</span></span>
<span id="cb104-188"><a href="#cb104-188" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span><span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>), ]</span>
<span id="cb104-189"><a href="#cb104-189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-190"><a href="#cb104-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-191"><a href="#cb104-191" aria-hidden="true" tabindex="-1"></a>My goal is to study the correlation of other features with these two disorder columns and NA values in these target variables would make that particular patient record useless. Hence I am removing rows which have null values in these target columns.</span>
<span id="cb104-192"><a href="#cb104-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-195"><a href="#cb104-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-196"><a href="#cb104-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping specific columns from the dataset</span></span>
<span id="cb104-197"><a href="#cb104-197" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">select</span>(data, <span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Patient Id</span><span class="st">`</span>, <span class="st">`</span><span class="at">Family Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Institute Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Patient First Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Father's name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Place of birth</span><span class="st">`</span>, <span class="st">`</span><span class="at">Location of Institute</span><span class="st">`</span>))</span>
<span id="cb104-198"><a href="#cb104-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-199"><a href="#cb104-199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-200"><a href="#cb104-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-201"><a href="#cb104-201" aria-hidden="true" tabindex="-1"></a>Dropping features that do not affect the type of genetic disorder or subclass in any way and hence analyzing them isn't of any use.</span>
<span id="cb104-202"><a href="#cb104-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-203"><a href="#cb104-203" aria-hidden="true" tabindex="-1"></a>Pre-processing plan: Task 3 and Specific Questions 1, 2 and 3</span>
<span id="cb104-204"><a href="#cb104-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-207"><a href="#cb104-207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-208"><a href="#cb104-208" aria-hidden="true" tabindex="-1"></a><span class="co"># Checking unique values in 'Test 1', 'Test 2', 'Test 3', 'Test 4', and 'Test 5' columns</span></span>
<span id="cb104-209"><a href="#cb104-209" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>)</span>
<span id="cb104-210"><a href="#cb104-210" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>)</span>
<span id="cb104-211"><a href="#cb104-211" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>)</span>
<span id="cb104-212"><a href="#cb104-212" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>)</span>
<span id="cb104-213"><a href="#cb104-213" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>)</span>
<span id="cb104-214"><a href="#cb104-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-215"><a href="#cb104-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-218"><a href="#cb104-218" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-219"><a href="#cb104-219" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with -99.0 in 'Test 1', 'Test 2', 'Test 3', 'Test 4', and 'Test 5' columns</span></span>
<span id="cb104-220"><a href="#cb104-220" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>)</span>
<span id="cb104-221"><a href="#cb104-221" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>)</span>
<span id="cb104-222"><a href="#cb104-222" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>)</span>
<span id="cb104-223"><a href="#cb104-223" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>)</span>
<span id="cb104-224"><a href="#cb104-224" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>), <span class="sc">-</span><span class="fl">99.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>)</span>
<span id="cb104-225"><a href="#cb104-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-226"><a href="#cb104-226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-227"><a href="#cb104-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-228"><a href="#cb104-228" aria-hidden="true" tabindex="-1"></a>Checking the unique values in the test results columns and replacing the missing values with an arbitrary negative value to identify them during future analysis.</span>
<span id="cb104-229"><a href="#cb104-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-232"><a href="#cb104-232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-233"><a href="#cb104-233" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `Patient Age`</span></span>
<span id="cb104-234"><a href="#cb104-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-235"><a href="#cb104-235" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"Patient age"</span>)) <span class="sc">+</span></span>
<span id="cb104-236"><a href="#cb104-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb104-237"><a href="#cb104-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-238"><a href="#cb104-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Patient Age"</span>) <span class="sc">+</span></span>
<span id="cb104-239"><a href="#cb104-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"red"</span>, <span class="at">labels =</span> <span class="st">"Patient age"</span>) <span class="sc">+</span></span>
<span id="cb104-240"><a href="#cb104-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-241"><a href="#cb104-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-242"><a href="#cb104-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-245"><a href="#cb104-245" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-246"><a href="#cb104-246" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `Blood cell count (mcL)` </span></span>
<span id="cb104-247"><a href="#cb104-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-248"><a href="#cb104-248" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"Blood cell count (mcL)"</span>)) <span class="sc">+</span></span>
<span id="cb104-249"><a href="#cb104-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb104-250"><a href="#cb104-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-251"><a href="#cb104-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb104-252"><a href="#cb104-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"blue"</span>, <span class="at">labels =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb104-253"><a href="#cb104-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-254"><a href="#cb104-254" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-255"><a href="#cb104-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-258"><a href="#cb104-258" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-259"><a href="#cb104-259" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plots for `White Blood cell count (thousand per microliter)` </span></span>
<span id="cb104-260"><a href="#cb104-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-261"><a href="#cb104-261" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>)) <span class="sc">+</span></span>
<span id="cb104-262"><a href="#cb104-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb104-263"><a href="#cb104-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Value"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-264"><a href="#cb104-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of White Blood cell count (thousand per microliter)"</span>) <span class="sc">+</span></span>
<span id="cb104-265"><a href="#cb104-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">"black"</span>, <span class="at">labels =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>) <span class="sc">+</span></span>
<span id="cb104-266"><a href="#cb104-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-267"><a href="#cb104-267" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-268"><a href="#cb104-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-269"><a href="#cb104-269" aria-hidden="true" tabindex="-1"></a>I can see that Patient age is uniformly distributed from 0-14 verifying the summary statistic that our dataset includes children in that particular age range only. Blood cell count has a normal distribution around 4.9mcL showing that most children have their blood cell count between 4mcL and 5.5mcL with majority of the children having the mean blood cell count of about 4.9mcL. I also observed that the White blood cell count is somewhat uniformly distributed between 3 and 12 thousand per microliter telling us all children have varying White blood cell count which reflects in their varying strength of immune system.</span>
<span id="cb104-270"><a href="#cb104-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-273"><a href="#cb104-273" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-274"><a href="#cb104-274" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Blood cell count (mcL)` against `White Blood cell count (thousand per #microliter)`</span></span>
<span id="cb104-275"><a href="#cb104-275" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-276"><a href="#cb104-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-277"><a href="#cb104-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"White Blood cell count (thousand per microliter)"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb104-278"><a href="#cb104-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"White Blood cell count vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb104-279"><a href="#cb104-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-280"><a href="#cb104-280" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-281"><a href="#cb104-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-282"><a href="#cb104-282" aria-hidden="true" tabindex="-1"></a>From the scatter plot, it looks like there are lot of outliers for the white blood cells. I can see that the blood cell count and white blood cell count do not have a strong positive or negative correlation, therefore the symptoms of in the blood sample are probably not caused by the white blood cells(leukocytes).</span>
<span id="cb104-283"><a href="#cb104-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-286"><a href="#cb104-286" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-287"><a href="#cb104-287" aria-hidden="true" tabindex="-1"></a><span class="co"># Create scatter plot of `Blood cell count (mcL)` against `Patient Age`</span></span>
<span id="cb104-288"><a href="#cb104-288" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-289"><a href="#cb104-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-290"><a href="#cb104-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb104-291"><a href="#cb104-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient Age"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb104-292"><a href="#cb104-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Patient Age vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb104-293"><a href="#cb104-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-294"><a href="#cb104-294" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-295"><a href="#cb104-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-298"><a href="#cb104-298" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-299"><a href="#cb104-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Create scatter plot of `White Blood cell count (thousand per microliter)` against `Patient Age`</span></span>
<span id="cb104-300"><a href="#cb104-300" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-301"><a href="#cb104-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-302"><a href="#cb104-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb104-303"><a href="#cb104-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient Age"</span>, <span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>) <span class="sc">+</span></span>
<span id="cb104-304"><a href="#cb104-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Patient Age vs. Blood cell count"</span>) <span class="sc">+</span></span>
<span id="cb104-305"><a href="#cb104-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-306"><a href="#cb104-306" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-307"><a href="#cb104-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-308"><a href="#cb104-308" aria-hidden="true" tabindex="-1"></a>From the scatter plots, I can see that the patients are uniformly distributed over the range of blood cell counts and white blood cell counts giving us the idea that there is very little to no correlation between the Cell count and Age of the patient.</span>
<span id="cb104-309"><a href="#cb104-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-312"><a href="#cb104-312" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-313"><a href="#cb104-313" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean values of `White Blood cell count (thousand per microliter)`,`Blood cell count #(mcL)`  and `Patient Age`</span></span>
<span id="cb104-314"><a href="#cb104-314" aria-hidden="true" tabindex="-1"></a>mean_wbc <span class="ot">&lt;-</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb104-315"><a href="#cb104-315" aria-hidden="true" tabindex="-1"></a>mean_bc <span class="ot">&lt;-</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb104-316"><a href="#cb104-316" aria-hidden="true" tabindex="-1"></a>mean_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb104-317"><a href="#cb104-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-318"><a href="#cb104-318" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean values</span></span>
<span id="cb104-319"><a href="#cb104-319" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>), mean_wbc, data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>)</span>
<span id="cb104-320"><a href="#cb104-320" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>), mean_bc, data<span class="sc">$</span><span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>)</span>
<span id="cb104-321"><a href="#cb104-321" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>), mean_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Patient Age</span><span class="st">`</span>)</span>
<span id="cb104-322"><a href="#cb104-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-323"><a href="#cb104-323" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-324"><a href="#cb104-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-325"><a href="#cb104-325" aria-hidden="true" tabindex="-1"></a>As the cell count and age are important features, I am replacing the missing cell counts with the mean value and the missing age with floor of the mean value since age needs to be a round number.</span>
<span id="cb104-326"><a href="#cb104-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-329"><a href="#cb104-329" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-330"><a href="#cb104-330" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Mother's age` against `Father's age`</span></span>
<span id="cb104-331"><a href="#cb104-331" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-332"><a href="#cb104-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-333"><a href="#cb104-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb104-334"><a href="#cb104-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Father's age vs. Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb104-335"><a href="#cb104-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-336"><a href="#cb104-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-337"><a href="#cb104-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-338"><a href="#cb104-338" aria-hidden="true" tabindex="-1"></a>I wanted to see if there is any correlation between the parents age which is clearly not the case from the views of the scatter plot.</span>
<span id="cb104-339"><a href="#cb104-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-342"><a href="#cb104-342" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-343"><a href="#cb104-343" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot with fitted normal distribution for Mother's age</span></span>
<span id="cb104-344"><a href="#cb104-344" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-345"><a href="#cb104-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb104-346"><a href="#cb104-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">sd =</span> <span class="fu">sd</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb104-347"><a href="#cb104-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Mother's age"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-348"><a href="#cb104-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb104-349"><a href="#cb104-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-350"><a href="#cb104-350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-351"><a href="#cb104-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-354"><a href="#cb104-354" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-355"><a href="#cb104-355" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot with fitted normal distribution for Father's age</span></span>
<span id="cb104-356"><a href="#cb104-356" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-357"><a href="#cb104-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb104-358"><a href="#cb104-358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">sd =</span> <span class="fu">sd</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb104-359"><a href="#cb104-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-360"><a href="#cb104-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of Father's age"</span>) <span class="sc">+</span></span>
<span id="cb104-361"><a href="#cb104-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-362"><a href="#cb104-362" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-363"><a href="#cb104-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-364"><a href="#cb104-364" aria-hidden="true" tabindex="-1"></a>Both the parents age follows a normal distribution  being symmetric about their mean value. This shows that majority of fathers are about 43 years old and majority of the mothers are about 35 years old.</span>
<span id="cb104-365"><a href="#cb104-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-368"><a href="#cb104-368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-369"><a href="#cb104-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean values</span></span>
<span id="cb104-370"><a href="#cb104-370" aria-hidden="true" tabindex="-1"></a>mean_mother_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb104-371"><a href="#cb104-371" aria-hidden="true" tabindex="-1"></a>mean_father_age <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb104-372"><a href="#cb104-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-373"><a href="#cb104-373" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean values</span></span>
<span id="cb104-374"><a href="#cb104-374" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>), mean_mother_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)</span>
<span id="cb104-375"><a href="#cb104-375" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>), mean_father_age, data<span class="sc">$</span><span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)</span>
<span id="cb104-376"><a href="#cb104-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-377"><a href="#cb104-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-378"><a href="#cb104-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-379"><a href="#cb104-379" aria-hidden="true" tabindex="-1"></a>I am replacing the missing values in Father's and Mother's age with the  floor of the mean value since age needs to be a round number.</span>
<span id="cb104-380"><a href="#cb104-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-383"><a href="#cb104-383" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-384"><a href="#cb104-384" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating density plot of `No. of previous abortion`</span></span>
<span id="cb104-385"><a href="#cb104-385" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">y =</span> ..density..)) <span class="sc">+</span></span>
<span id="cb104-386"><a href="#cb104-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb104-387"><a href="#cb104-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb104-388"><a href="#cb104-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of No. of previous abortion"</span>) <span class="sc">+</span></span>
<span id="cb104-389"><a href="#cb104-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-390"><a href="#cb104-390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-391"><a href="#cb104-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-394"><a href="#cb104-394" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-395"><a href="#cb104-395" aria-hidden="true" tabindex="-1"></a><span class="co">#Checking unique values in `No. of previous abortion`</span></span>
<span id="cb104-396"><a href="#cb104-396" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>)</span>
<span id="cb104-397"><a href="#cb104-397" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-398"><a href="#cb104-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-399"><a href="#cb104-399" aria-hidden="true" tabindex="-1"></a>I wanted to learn about the <span class="in">`No. of previous abortion`</span> variable and from its density distribution, it is surprising to see that there are many children who have had 1,2,3 and even 4 abortions already. I want to check their Mothers age to get more insights.</span>
<span id="cb104-400"><a href="#cb104-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-403"><a href="#cb104-403" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-404"><a href="#cb104-404" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating scatter plot of `Mother's age` against `No. of previous abortion`</span></span>
<span id="cb104-405"><a href="#cb104-405" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-406"><a href="#cb104-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-407"><a href="#cb104-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Mother's age"</span>) <span class="sc">+</span></span>
<span id="cb104-408"><a href="#cb104-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Mother's age vs. No. of previous abortion"</span>) <span class="sc">+</span></span>
<span id="cb104-409"><a href="#cb104-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-410"><a href="#cb104-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-411"><a href="#cb104-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-412"><a href="#cb104-412" aria-hidden="true" tabindex="-1"></a>Wow, there are cases where patients with just 18 year old mothers have had 4 abortions. I want to look into the Father's age of these patients.</span>
<span id="cb104-413"><a href="#cb104-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-416"><a href="#cb104-416" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-417"><a href="#cb104-417" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering data for Mother's age equal to 18</span></span>
<span id="cb104-418"><a href="#cb104-418" aria-hidden="true" tabindex="-1"></a>filtered_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(data, <span class="st">`</span><span class="at">Mother's age</span><span class="st">`</span> <span class="sc">==</span> <span class="dv">18</span>)</span>
<span id="cb104-419"><a href="#cb104-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-420"><a href="#cb104-420" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating count Bar graph of Father's age for these records</span></span>
<span id="cb104-421"><a href="#cb104-421" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Father's age</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-422"><a href="#cb104-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-423"><a href="#cb104-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Father's age"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb104-424"><a href="#cb104-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Father's age for 18 year old mother"</span>) <span class="sc">+</span></span>
<span id="cb104-425"><a href="#cb104-425" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb104-426"><a href="#cb104-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb104-427"><a href="#cb104-427" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-428"><a href="#cb104-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-429"><a href="#cb104-429" aria-hidden="true" tabindex="-1"></a>I can see that majority of fathers of these patients who have a 18 year old mother and have had a few abortions are about 41 years old.</span>
<span id="cb104-430"><a href="#cb104-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-433"><a href="#cb104-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-434"><a href="#cb104-434" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculating mean value of `No. of previous abortion`</span></span>
<span id="cb104-435"><a href="#cb104-435" aria-hidden="true" tabindex="-1"></a>mean_previous_abortion <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">mean</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb104-436"><a href="#cb104-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-437"><a href="#cb104-437" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values with calculated mean value</span></span>
<span id="cb104-438"><a href="#cb104-438" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>), mean_previous_abortion, data<span class="sc">$</span><span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>)</span>
<span id="cb104-439"><a href="#cb104-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-440"><a href="#cb104-440" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-441"><a href="#cb104-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-442"><a href="#cb104-442" aria-hidden="true" tabindex="-1"></a>I am replacing the missing values in No. of previous abortion with the  floor of the mean value since the variable needs to be a round number.</span>
<span id="cb104-443"><a href="#cb104-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-446"><a href="#cb104-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-447"><a href="#cb104-447" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA values of Symptoms columns with 0.0</span></span>
<span id="cb104-448"><a href="#cb104-448" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 1</span><span class="st">`</span>)</span>
<span id="cb104-449"><a href="#cb104-449" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 2</span><span class="st">`</span>)</span>
<span id="cb104-450"><a href="#cb104-450" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 3</span><span class="st">`</span>)</span>
<span id="cb104-451"><a href="#cb104-451" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 4</span><span class="st">`</span>)</span>
<span id="cb104-452"><a href="#cb104-452" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span>), <span class="fl">0.0</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Symptom 5</span><span class="st">`</span>)</span>
<span id="cb104-453"><a href="#cb104-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-454"><a href="#cb104-454" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-455"><a href="#cb104-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-456"><a href="#cb104-456" aria-hidden="true" tabindex="-1"></a>I am replacing the missing values in the Symptoms columns with 0 since 0 represents no symptoms and 1 represents that symptoms are present and when the symptoms aren't mentioned, it is safe to assume that they have no symptoms. </span>
<span id="cb104-457"><a href="#cb104-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-460"><a href="#cb104-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-461"><a href="#cb104-461" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating count Bar graphs for Categorical features</span></span>
<span id="cb104-462"><a href="#cb104-462" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-463"><a href="#cb104-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb104-464"><a href="#cb104-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Maternal gene"</span>) <span class="sc">+</span></span>
<span id="cb104-465"><a href="#cb104-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-466"><a href="#cb104-466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-467"><a href="#cb104-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-468"><a href="#cb104-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-471"><a href="#cb104-471" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-472"><a href="#cb104-472" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-473"><a href="#cb104-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb104-474"><a href="#cb104-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Respiratory Rate (breaths/min)"</span>) <span class="sc">+</span></span>
<span id="cb104-475"><a href="#cb104-475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-476"><a href="#cb104-476" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-477"><a href="#cb104-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-478"><a href="#cb104-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-481"><a href="#cb104-481" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-482"><a href="#cb104-482" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-483"><a href="#cb104-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"green"</span>) <span class="sc">+</span></span>
<span id="cb104-484"><a href="#cb104-484" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Heart Rate (rates/min)"</span>) <span class="sc">+</span></span>
<span id="cb104-485"><a href="#cb104-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-486"><a href="#cb104-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-487"><a href="#cb104-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-488"><a href="#cb104-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-491"><a href="#cb104-491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-492"><a href="#cb104-492" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-493"><a href="#cb104-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Parental consent</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"orange"</span>) <span class="sc">+</span></span>
<span id="cb104-494"><a href="#cb104-494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Parental consent"</span>) <span class="sc">+</span></span>
<span id="cb104-495"><a href="#cb104-495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-496"><a href="#cb104-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-497"><a href="#cb104-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-498"><a href="#cb104-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-501"><a href="#cb104-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-502"><a href="#cb104-502" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-503"><a href="#cb104-503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"purple"</span>) <span class="sc">+</span></span>
<span id="cb104-504"><a href="#cb104-504" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Follow-up"</span>) <span class="sc">+</span></span>
<span id="cb104-505"><a href="#cb104-505" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-506"><a href="#cb104-506" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-507"><a href="#cb104-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-508"><a href="#cb104-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-511"><a href="#cb104-511" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-512"><a href="#cb104-512" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-513"><a href="#cb104-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"pink"</span>) <span class="sc">+</span></span>
<span id="cb104-514"><a href="#cb104-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Folic acid details (peri-conceptional)"</span>) <span class="sc">+</span></span>
<span id="cb104-515"><a href="#cb104-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-516"><a href="#cb104-516" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-517"><a href="#cb104-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-518"><a href="#cb104-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-521"><a href="#cb104-521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-522"><a href="#cb104-522" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-523"><a href="#cb104-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"brown"</span>) <span class="sc">+</span></span>
<span id="cb104-524"><a href="#cb104-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: H/O radiation exposure (x-ray)"</span>) <span class="sc">+</span></span>
<span id="cb104-525"><a href="#cb104-525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-526"><a href="#cb104-526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-527"><a href="#cb104-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-528"><a href="#cb104-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-531"><a href="#cb104-531" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-532"><a href="#cb104-532" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-533"><a href="#cb104-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb104-534"><a href="#cb104-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: H/O substance abuse"</span>) <span class="sc">+</span></span>
<span id="cb104-535"><a href="#cb104-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-536"><a href="#cb104-536" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-537"><a href="#cb104-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-538"><a href="#cb104-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-541"><a href="#cb104-541" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-542"><a href="#cb104-542" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-543"><a href="#cb104-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"cyan"</span>) <span class="sc">+</span></span>
<span id="cb104-544"><a href="#cb104-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: Assisted conception IVF/ART"</span>) <span class="sc">+</span></span>
<span id="cb104-545"><a href="#cb104-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-546"><a href="#cb104-546" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-547"><a href="#cb104-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-548"><a href="#cb104-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-551"><a href="#cb104-551" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-552"><a href="#cb104-552" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb104-553"><a href="#cb104-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>), <span class="at">fill =</span> <span class="st">"yellow"</span>) <span class="sc">+</span></span>
<span id="cb104-554"><a href="#cb104-554" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Count Plot: History of anomalies in previous pregnancies"</span>) <span class="sc">+</span></span>
<span id="cb104-555"><a href="#cb104-555" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-556"><a href="#cb104-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-557"><a href="#cb104-557" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-558"><a href="#cb104-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-559"><a href="#cb104-559" aria-hidden="true" tabindex="-1"></a>From the Bar graphs of the counts of these categorical features, I can see which sub category of each variable accounts for the majority percentage. Since there are lot of parental consent, I will assume that tests conducted due to parents consent and there's no way a kid will check up on his own. </span>
<span id="cb104-560"><a href="#cb104-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-563"><a href="#cb104-563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-564"><a href="#cb104-564" aria-hidden="true" tabindex="-1"></a><span class="co"># Dropping Parental consent column</span></span>
<span id="cb104-565"><a href="#cb104-565" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">select</span>(data, <span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Parental consent</span><span class="st">`</span>))</span>
<span id="cb104-566"><a href="#cb104-566" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-567"><a href="#cb104-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-568"><a href="#cb104-568" aria-hidden="true" tabindex="-1"></a>Dropping the parental consent since most of them require it and the feature does not have influence on the type of genetic disorder.</span>
<span id="cb104-569"><a href="#cb104-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-572"><a href="#cb104-572" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-573"><a href="#cb104-573" aria-hidden="true" tabindex="-1"></a><span class="co"># Checking unique values in columns</span></span>
<span id="cb104-574"><a href="#cb104-574" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>)</span>
<span id="cb104-575"><a href="#cb104-575" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span>
<span id="cb104-576"><a href="#cb104-576" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span>
<span id="cb104-577"><a href="#cb104-577" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>)</span>
<span id="cb104-578"><a href="#cb104-578" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>)</span>
<span id="cb104-579"><a href="#cb104-579" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>)</span>
<span id="cb104-580"><a href="#cb104-580" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>)</span>
<span id="cb104-581"><a href="#cb104-581" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>)</span>
<span id="cb104-582"><a href="#cb104-582" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>)</span>
<span id="cb104-583"><a href="#cb104-583" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>)</span>
<span id="cb104-584"><a href="#cb104-584" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Gender</span><span class="st">`</span>)</span>
<span id="cb104-585"><a href="#cb104-585" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>)</span>
<span id="cb104-586"><a href="#cb104-586" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>)</span>
<span id="cb104-587"><a href="#cb104-587" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>)</span>
<span id="cb104-588"><a href="#cb104-588" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>)</span>
<span id="cb104-589"><a href="#cb104-589" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>)</span>
<span id="cb104-590"><a href="#cb104-590" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span>Status)</span>
<span id="cb104-591"><a href="#cb104-591" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>)</span>
<span id="cb104-592"><a href="#cb104-592" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>)</span>
<span id="cb104-593"><a href="#cb104-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-594"><a href="#cb104-594" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-595"><a href="#cb104-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-598"><a href="#cb104-598" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-599"><a href="#cb104-599" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA with "No" in specific columns</span></span>
<span id="cb104-600"><a href="#cb104-600" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>)</span>
<span id="cb104-601"><a href="#cb104-601" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Folic acid details (peri-conceptional)</span><span class="st">`</span>)</span>
<span id="cb104-602"><a href="#cb104-602" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>)</span>
<span id="cb104-603"><a href="#cb104-603" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>)</span>
<span id="cb104-604"><a href="#cb104-604" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Assisted conception IVF/ART</span><span class="st">`</span>)</span>
<span id="cb104-605"><a href="#cb104-605" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">History of anomalies in previous pregnancies</span><span class="st">`</span>)</span>
<span id="cb104-606"><a href="#cb104-606" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>)</span>
<span id="cb104-607"><a href="#cb104-607" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">H/O serious maternal illness</span><span class="st">`</span>)</span>
<span id="cb104-608"><a href="#cb104-608" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Inherited from father</span><span class="st">`</span>)</span>
<span id="cb104-609"><a href="#cb104-609" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>)</span>
<span id="cb104-610"><a href="#cb104-610" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Genes in mother's side</span><span class="st">`</span>)</span>
<span id="cb104-611"><a href="#cb104-611" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>), <span class="st">"No"</span>, data<span class="sc">$</span><span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>)</span>
<span id="cb104-612"><a href="#cb104-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-613"><a href="#cb104-613" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-614"><a href="#cb104-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-617"><a href="#cb104-617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-618"><a href="#cb104-618" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate mode</span></span>
<span id="cb104-619"><a href="#cb104-619" aria-hidden="true" tabindex="-1"></a>Mode <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb104-620"><a href="#cb104-620" aria-hidden="true" tabindex="-1"></a>  ux <span class="ot">&lt;-</span> <span class="fu">unique</span>(x)</span>
<span id="cb104-621"><a href="#cb104-621" aria-hidden="true" tabindex="-1"></a>  ux[<span class="fu">which.max</span>(<span class="fu">tabulate</span>(<span class="fu">match</span>(x, ux)))]</span>
<span id="cb104-622"><a href="#cb104-622" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb104-623"><a href="#cb104-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-624"><a href="#cb104-624" aria-hidden="true" tabindex="-1"></a><span class="co"># Filling NA with the most frequent value in specific columns</span></span>
<span id="cb104-625"><a href="#cb104-625" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span>
<span id="cb104-626"><a href="#cb104-626" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span>
<span id="cb104-627"><a href="#cb104-627" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Follow-up</span><span class="st">`</span>)</span>
<span id="cb104-628"><a href="#cb104-628" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Gender <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span>Gender), <span class="fu">Mode</span>(data<span class="sc">$</span>Gender), data<span class="sc">$</span>Gender)</span>
<span id="cb104-629"><a href="#cb104-629" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Birth defects</span><span class="st">`</span>)</span>
<span id="cb104-630"><a href="#cb104-630" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>), <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>), data<span class="sc">$</span><span class="st">`</span><span class="at">Blood test result</span><span class="st">`</span>)</span>
<span id="cb104-631"><a href="#cb104-631" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(data<span class="sc">$</span>Status), <span class="fu">Mode</span>(data<span class="sc">$</span>Status), data<span class="sc">$</span>Status)</span>
<span id="cb104-632"><a href="#cb104-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-633"><a href="#cb104-633" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-634"><a href="#cb104-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-635"><a href="#cb104-635" aria-hidden="true" tabindex="-1"></a>I checked the values in these columns and replaced the missing values in the categorical features with "No" since that would be most apt if the illness wasn't mentioned and with value of the most frequent value(mode) in the numerical features since these variables have a partial but not so strong impact on the genetic disorder. </span>
<span id="cb104-636"><a href="#cb104-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-639"><a href="#cb104-639" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-640"><a href="#cb104-640" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing -99 with 0 in 'Test 1', 'Test 2', 'Test 3', 'Test 4', 'Test 5'</span></span>
<span id="cb104-641"><a href="#cb104-641" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 1</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb104-642"><a href="#cb104-642" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 2</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb104-643"><a href="#cb104-643" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 3</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb104-644"><a href="#cb104-644" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 4</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb104-645"><a href="#cb104-645" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Test 5</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb104-646"><a href="#cb104-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-647"><a href="#cb104-647" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing -99 in 'White Blood cell count (thousand per microliter)' with median</span></span>
<span id="cb104-648"><a href="#cb104-648" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">99</span>] <span class="ot">&lt;-</span> <span class="fu">median</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">White Blood cell count (thousand per microliter)</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb104-649"><a href="#cb104-649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-650"><a href="#cb104-650" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing '-99' with mode in 'Respiratory Rate (breaths/min)', 'Heart Rate (rates/min)'</span></span>
<span id="cb104-651"><a href="#cb104-651" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-99'</span>] <span class="ot">&lt;-</span> <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>)</span>
<span id="cb104-652"><a href="#cb104-652" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-99'</span>] <span class="ot">&lt;-</span> <span class="fu">Mode</span>(data<span class="sc">$</span><span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>)</span>
<span id="cb104-653"><a href="#cb104-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-654"><a href="#cb104-654" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'Normal (30-60)' with 'Normal' in 'Respiratory Rate (breaths/min)'</span></span>
<span id="cb104-655"><a href="#cb104-655" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'Normal (30-60)'</span>] <span class="ot">&lt;-</span> <span class="st">'Normal'</span></span>
<span id="cb104-656"><a href="#cb104-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-657"><a href="#cb104-657" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'No record' with 'Not available' in 'Birth asphyxia'</span></span>
<span id="cb104-658"><a href="#cb104-658" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'No record'</span>] <span class="ot">&lt;-</span> <span class="st">'Not available'</span></span>
<span id="cb104-659"><a href="#cb104-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-660"><a href="#cb104-660" aria-hidden="true" tabindex="-1"></a><span class="co"># Replacing 'None' and '-' with 'No' in 'Autopsy shows birth defect (if applicable)', 'H/O radiation exposure (x-ray)', 'H/O substance abuse'</span></span>
<span id="cb104-661"><a href="#cb104-661" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">Autopsy shows birth defect (if applicable)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span>
<span id="cb104-662"><a href="#cb104-662" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span>
<span id="cb104-663"><a href="#cb104-663" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span>[data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'None'</span> <span class="sc">|</span> data<span class="sc">$</span><span class="st">`</span><span class="at">H/O substance abuse</span><span class="st">`</span> <span class="sc">==</span> <span class="st">'-'</span>] <span class="ot">&lt;-</span> <span class="st">'No'</span></span>
<span id="cb104-664"><a href="#cb104-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-665"><a href="#cb104-665" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-666"><a href="#cb104-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-667"><a href="#cb104-667" aria-hidden="true" tabindex="-1"></a>I replaced arbitrary negative value(-99) in Test results feature with 0 since it implies that missing value equals a negative test which is the most appropriate to assume. I replaced the arbitrary value in White Blood cell count, Respiratory and heart rate with median and mode respectively as taking the central tendency value would make the most sense as these features are key for determining the type of genetic disorder. I replaced 'Normal (30-60)' with 'Normal', 'No record' with 'Not available', 'None' and '-' with 'No' in Respiratory Rate, Birth asphyxia, Autopsy shows birth defect, H/O radiation exposure and H/O substance abuse respectively to maintain uniformity within the sub categories. </span>
<span id="cb104-668"><a href="#cb104-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-669"><a href="#cb104-669" aria-hidden="true" tabindex="-1"></a>Specific Question 4</span>
<span id="cb104-670"><a href="#cb104-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-673"><a href="#cb104-673" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-674"><a href="#cb104-674" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the count plot for `Disorder Subclass`</span></span>
<span id="cb104-675"><a href="#cb104-675" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Disorder Subclass</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-676"><a href="#cb104-676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb104-677"><a href="#cb104-677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Disorder Subclass"</span>, <span class="at">x =</span> <span class="st">"Disorder Subclass"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb104-678"><a href="#cb104-678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb104-679"><a href="#cb104-679" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-680"><a href="#cb104-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-681"><a href="#cb104-681" aria-hidden="true" tabindex="-1"></a>From the Count plot for the subclasses of genetic disorder, I can say that most of the patients suffer from Leigh Syndrome while the least amount of children have Cancer.</span>
<span id="cb104-682"><a href="#cb104-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-683"><a href="#cb104-683" aria-hidden="true" tabindex="-1"></a>Specific Question 5</span>
<span id="cb104-684"><a href="#cb104-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-687"><a href="#cb104-687" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-688"><a href="#cb104-688" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Maternal gene and Genetic Disorder</span></span>
<span id="cb104-689"><a href="#cb104-689" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Maternal gene</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-690"><a href="#cb104-690" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-691"><a href="#cb104-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Maternal gene"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-692"><a href="#cb104-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Maternal gene vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-693"><a href="#cb104-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-694"><a href="#cb104-694" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-695"><a href="#cb104-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-698"><a href="#cb104-698" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-699"><a href="#cb104-699" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Paternal gene and Genetic Disorder</span></span>
<span id="cb104-700"><a href="#cb104-700" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Paternal gene</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-701"><a href="#cb104-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-702"><a href="#cb104-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Paternal gene"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-703"><a href="#cb104-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Paternal gene vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-704"><a href="#cb104-704" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-705"><a href="#cb104-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-706"><a href="#cb104-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-707"><a href="#cb104-707" aria-hidden="true" tabindex="-1"></a>I observe that the maternal gene equally affects patients and is does not determine the type of genetic disorder whereas the absence of the paternal gene is seen in more cases of genetic abnormalities as compared to the patients containing the paternal gene.</span>
<span id="cb104-708"><a href="#cb104-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-711"><a href="#cb104-711" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-712"><a href="#cb104-712" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Respiratory Rate (breaths/min) and Genetic Disorder</span></span>
<span id="cb104-713"><a href="#cb104-713" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Respiratory Rate (breaths/min)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-714"><a href="#cb104-714" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-715"><a href="#cb104-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Respiratory Rate (breaths/min)"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-716"><a href="#cb104-716" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Respiratory Rate (breaths/min) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-717"><a href="#cb104-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-718"><a href="#cb104-718" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-719"><a href="#cb104-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-722"><a href="#cb104-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-723"><a href="#cb104-723" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Heart Rate (rates/min and Genetic Disorder</span></span>
<span id="cb104-724"><a href="#cb104-724" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Heart Rate (rates/min</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-725"><a href="#cb104-725" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-726"><a href="#cb104-726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Heart Rate (rates/min"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-727"><a href="#cb104-727" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Heart Rate (rates/min) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-728"><a href="#cb104-728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-729"><a href="#cb104-729" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-730"><a href="#cb104-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-733"><a href="#cb104-733" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-734"><a href="#cb104-734" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Birth asphyxia and Genetic Disorder</span></span>
<span id="cb104-735"><a href="#cb104-735" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Birth asphyxia</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-736"><a href="#cb104-736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-737"><a href="#cb104-737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Birth asphyxia"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-738"><a href="#cb104-738" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Birth asphyxia vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-739"><a href="#cb104-739" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-740"><a href="#cb104-740" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-741"><a href="#cb104-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-742"><a href="#cb104-742" aria-hidden="true" tabindex="-1"></a>I see that patients with a normal Respiratory breathing rate are in larger proportion when it comes to having a genetic disorder and hence I think that Respiratory rate may not be a primary factor for determining presence of a genetic disease. In the same manner, Heart rate and Birth asphyxia can also be taken out of consideration for early prognosis of a genetic disorder.</span>
<span id="cb104-743"><a href="#cb104-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-746"><a href="#cb104-746" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-747"><a href="#cb104-747" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between H/O radiation exposure (x-ray) and Genetic Disorder</span></span>
<span id="cb104-748"><a href="#cb104-748" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">H/O radiation exposure (x-ray)</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-749"><a href="#cb104-749" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-750"><a href="#cb104-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"H/O radiation exposure (x-ray)"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-751"><a href="#cb104-751" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"H/O radiation exposure (x-ray) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-752"><a href="#cb104-752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-753"><a href="#cb104-753" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-754"><a href="#cb104-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-755"><a href="#cb104-755" aria-hidden="true" tabindex="-1"></a>I observe that people with no radiation exposure also suffer from a lot of genetic disorders compared to people exposed to x-rays, and this condition may also not be a part of predicting the presence of a genetic disease in the patient.</span>
<span id="cb104-756"><a href="#cb104-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-759"><a href="#cb104-759" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-760"><a href="#cb104-760" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between No. of previous abortion and Genetic Disorder</span></span>
<span id="cb104-761"><a href="#cb104-761" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">No. of previous abortion</span><span class="st">`</span>, <span class="at">fill =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-762"><a href="#cb104-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb104-763"><a href="#cb104-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"No. of previous abortion"</span>, <span class="at">y =</span> <span class="st">"Count"</span>, <span class="at">fill =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-764"><a href="#cb104-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"No. of previous abortion vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-765"><a href="#cb104-765" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-766"><a href="#cb104-766" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-767"><a href="#cb104-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-768"><a href="#cb104-768" aria-hidden="true" tabindex="-1"></a>From the correlation plot, patients with even a single previous abortion tend to have a higher chance of suffering from a genetic disorder, mainly Mitochondrial Inheritance disorders.</span>
<span id="cb104-769"><a href="#cb104-769" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-772"><a href="#cb104-772" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb104-773"><a href="#cb104-773" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the correlation graph between Blood cell count (mcL) and Genetic Disorder</span></span>
<span id="cb104-774"><a href="#cb104-774" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">y =</span> <span class="st">`</span><span class="at">Blood cell count (mcL)</span><span class="st">`</span>, <span class="at">x =</span> <span class="st">`</span><span class="at">Genetic Disorder</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb104-775"><a href="#cb104-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb104-776"><a href="#cb104-776" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Blood cell count (mcL)"</span>, <span class="at">x =</span> <span class="st">"Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-777"><a href="#cb104-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Blood cell count (mcL) vs. Genetic Disorder"</span>) <span class="sc">+</span></span>
<span id="cb104-778"><a href="#cb104-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb104-779"><a href="#cb104-779" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb104-780"><a href="#cb104-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-781"><a href="#cb104-781" aria-hidden="true" tabindex="-1"></a>From the graph, I see that patients with blood cell count within the full range suffer from genetic disorders which means that this feature may not be a key influence for causing genetic diseases in children.</span>
<span id="cb104-782"><a href="#cb104-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-783"><a href="#cb104-783" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 7. Conclusion and Discussion :</span></span>
<span id="cb104-784"><a href="#cb104-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-785"><a href="#cb104-785" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>The analysis of the dataset has provided valuable insights into the relationship between various features and the presence of genetic disorders. The first step was to remove rows with missing values in the target columns, as these records would not contribute to the analysis. Additionally, irrelevant features that did not affect the type of genetic disorder were dropped, streamlining the dataset for further examination. The examination of the test results columns revealed unique values and the replacement of missing values with an arbitrary negative value for future identification. The analysis of patient age, blood cell count, and white blood cell count showcased interesting patterns. Patient age was found to be uniformly distributed within the range of 0 to 14 years, indicating that the dataset primarily consists of children in this age group. Blood cell count exhibited a normal distribution, with the majority of children having counts between 4mcL and 5.5mcL. Similarly, white blood cell count showed a somewhat uniform distribution between 3 and 12 thousand per microliter, reflecting the varying strength of the immune system among children.</span>
<span id="cb104-786"><a href="#cb104-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-787"><a href="#cb104-787" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Scatter plots were employed to explore potential correlations between variables. The scatter plot of blood cell count and white blood cell count indicated no strong positive or negative correlation, suggesting that the symptoms observed in the blood sample may not be caused by variations in white blood cell count. Additionally, scatter plots of blood cell count versus patient age and parental age exhibited no discernible correlation, indicating that these factors may have limited influence on the occurrence of genetic disorders.</span>
<span id="cb104-788"><a href="#cb104-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-789"><a href="#cb104-789" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>The examination of parental age distributions revealed normal distributions for both fathers and mothers, with the majority of fathers being around 43 years old and mothers around 35 years old. Missing values in father's and mother's age were replaced with the floor of the mean value to maintain consistency and roundness in the age representation. Exploring the "No. of previous abortion" variable provided insights into the frequency of abortions among patients. It was observed that there were cases where patients with 18-year-old mothers had undergone multiple abortions. Furthermore, the examination of father's age in these cases revealed that the majority of fathers were around 41 years old.</span>
<span id="cb104-790"><a href="#cb104-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-791"><a href="#cb104-791" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Count plots were used to analyze categorical features, such as parental consent, and revealed the predominance of specific categories. As a result, the parental consent feature was dropped, as it was determined to have little influence on the type of genetic disorder. Missing values in categorical features were replaced with "No" to represent the absence of the mentioned condition. Furthermore, specific values such as "Normal (30-60)" were replaced with simplified representations like "Normal" for consistency.</span>
<span id="cb104-792"><a href="#cb104-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-793"><a href="#cb104-793" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Count plots of the subclasses of genetic disorders highlighted the distribution of disorders within the dataset. Leigh Syndrome was found to be the most prevalent disorder, while Cancer accounted for the fewest cases. The analysis of the maternal gene revealed that it did not significantly influence the type of genetic disorder, whereas the absence of the paternal gene was more prevalent in cases of genetic abnormalities.</span>
<span id="cb104-794"><a href="#cb104-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-795"><a href="#cb104-795" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>The examination of features such as respiratory rate, heart rate, radiation exposure, and birth asphyxia indicated that these factors may not be primary indicators of genetic disorders. Similarly, the correlation plot showed that patients with previous abortions tended to have a higher likelihood of suffering from genetic disorders, particularly those related to mitochondrial inheritance. Lastly, the scatter plot of blood cell count versus genetic disorders suggested that blood cell count may not be a key determining factor in the presence of genetic diseases.</span>
<span id="cb104-796"><a href="#cb104-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-797"><a href="#cb104-797" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>In conclusion, the analysis of the dataset provided valuable insights into the relationship between various features and the occurrence of genetic disorders. The examination of unique values, data distributions, scatter plots, and count plots enabled a deeper understanding of the dataset and facilitated the identification of influential factors. These findings contribute to the field of genetic disorder research and can potentially aid in early prognosis and understanding the underlying mechanisms of these disorders</span>
<span id="cb104-798"><a href="#cb104-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-799"><a href="#cb104-799" aria-hidden="true" tabindex="-1"></a><span class="fu">## Part 8. Bibliography :</span></span>
<span id="cb104-800"><a href="#cb104-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-801"><a href="#cb104-801" aria-hidden="true" tabindex="-1"></a>Dataset : https://www.kaggle.com/datasets/aibuzz/predict-the-genetic-disorders-datasetof-genomes</span>
<span id="cb104-802"><a href="#cb104-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-803"><a href="#cb104-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-804"><a href="#cb104-804" aria-hidden="true" tabindex="-1"></a>Programming Language : R</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>